---
title: "Land Crab Gut Microbiomes"
author: "Cassie Ettinger"
date: "`r Sys.Date()`"
#output: pdf_document
output:  github_document
---

Project: Unlocking the terrestrial realm: Are microbial symbioses the key to land crab terrestrial invasions?
Collaborators: Cassie Ettinger, Victoria Watson-Zink, Laetitia Wilkins

What role does the gut microbiome of land crabs play in their ability to digest recalcitrant plant matter? To address this question, this project (1) characterizes the microbial community inhabiting the gut tissues of land crabs from lineages that have invaded terrestrial habitats; (2) asess whether there is an evolutionary or phylogenetic context to these microbial associations; and (3) compare the putative functions of gut microbiomes in marine and terrestrial crabs. 

ChatGPT 3.5 (https://openai.com) was asked to suggest improvements to code comments. These  suggestions were edited and incorporated as appropriate to help streamline documentation for readability. 

```{r setup, include=FALSE}
#Setup RMarkdown doc; not included in knit
library(knitr)
knitr::opts_chunk$set(echo = TRUE)
opts_chunk$set(fig.width = 6, fig.height = 4)
opts_chunk$set(tidy.opts=list(width.cutoff=60),tidy=TRUE)
knitr::opts_chunk$set(cache.extra = rand_seed)

```

### Load in required libraries
```{r libraries, message = FALSE, warning=FALSE}

# Load required R packages
library(tidyverse)
library(phyloseq)
library(vroom)
library(microbiome)
library(patchwork)
library(vegan)
library(pairwiseAdonis)
library(ggtext)
library(decontam)
library(FSA)

# Load custom R script with miscellaneous functions
source(file = "scripts/R_scripts/misc_functions.R")


```

### Import metagenomic coverage information from coverM and prepare for analysis
```{r coverm}

# Load metadata from "metadata_crabs.csv" and convert it to a sample_data object
metadata <- vroom("data/metadata_crabs.csv")
meta <- sample_data(metadata)
rownames(meta) <- meta$FilePrefix

# Load coverM data and preprocess it to split into RPKM metric 
coverm_mag <- vroom("data/MAGs/coverm/coverm_results_mincov0_aln75_id95.tsv") %>%
  pivot_longer(cols = -c(1), names_to="Sample") %>%
  pivot_wider(names_from = Genome,values_from = value) %>%
  separate(Sample, into = c("Sample", "Abundance"), sep=" ", extra = "merge") %>%
  mutate(Sample = gsub("_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz", "", Sample))

# Filter the coverM data to include only RPKM 
rpkm_mag <- coverm_mag %>%
  filter(Abundance == "RPKM")

# Prepare coverM data into otu_table format for phyloseq
otu_coverM <- as.data.frame(rpkm_mag[-c(2:3)])
row.names(otu_coverM) <- otu_coverM$Sample
otu_coverM <- otu_coverM[-c(1)]
otu_coverM_table <- otu_table(otu_coverM, taxa_are_rows = FALSE)

# Load MAG taxonomy data and clean up taxonomy names
gtdbtk <- vroom("data/MAGs/CVL_final_bin_set_20230516.csv") %>%
  separate("classification", into = c("domain", "phylum", "class", "order", "family", "genus", "species"), sep=";") %>%
  mutate(species = gsub(".__", "",species),
         genus = gsub(".__", "",genus),
         family = gsub(".__", "",family),
         order = gsub(".__", "",order),
         class = gsub(".__", "",class),
         phylum = gsub(".__", "",phylum),
         domain = gsub(".__", "",domain)) %>%
  mutate(across(where(is.character), ~na_if(., ""))) %>%
  mutate(species = ifelse(is.na(species), (ifelse(is.na(genus), ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family)), as.character(genus))), as.character(species))) %>%
  mutate(genus = ifelse(is.na(genus), ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family)), as.character(genus))) %>%
  mutate(family = ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family))) %>%
  mutate(order = ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order))) %>%
  mutate(class = ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class))) %>%
  mutate(phylum = ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum))) 

# Prepare MAG taxonomy data into tax_table format for phyloseq
tax.mag <- as.data.frame(gtdbtk[c(1, 11:17)])
row.names(tax.mag) <- tax.mag$bin_name
tax.mag <- tax.mag[-c(1)]
tax_table_mag <- tax_table(as.matrix(tax.mag))

# Create a phyloseq object with the coverM data and its taxonomy
physeq_coverM <- phyloseq(otu_coverM_table, tax_table_mag, meta)

physeq_coverM


```

### Import kraken2/braken read-based information and prepare for analysis
```{r krak}

# Prior to R need to obtain the full taxonomy string for each taxid output by kraken and bind to our data

# cut -f2 species_bracken.tsv | tail -n +2 | ete3 ncbiquery --info > species_taxonomy.txt
# /opt/miniconda3/envs/ete3/lib/python3.6/site-packages/ete3-3.1.2-py3.7.egg/ete3/ncbi_taxonomy/ncbiquery.py:243: UserWarning: taxid 2715211 was translated into 2758038
# /opt/miniconda3/envs/ete3/lib/python3.6/site-packages/ete3-3.1.2-py3.7.egg/ete3/ncbi_taxonomy/ncbiquery.py:243: UserWarning: taxid 2918672 was translated into 2917790

# cut -f2 genus_bracken.tsv | tail -n +2 | ete3 ncbiquery --info > genus_taxonomy.txt
# cut -f2 family_bracken.tsv | tail -n +2 | ete3 ncbiquery --info > family_taxonomy.txt
# cut -f2 phylum_bracken.tsv | tail -n +2 | ete3 ncbiquery --info > phylum_taxonomy.txt

# Import taxonomy data from NCBI/ete3
tax2name <- vroom("data/braken_results/species_taxonomy.txt")

# Clean up taxonomy levels that contain incorrect strings
tax2name <- tax2name %>% 
  mutate_at("Named Lineage", str_replace, "Bacteroidetes/Chlorobi group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Terrabacteria group,", "") %>%
  mutate_at("Named Lineage", str_replace, "delta/epsilon subdivisions,", "") %>%
  mutate_at("Named Lineage", str_replace, "Cystobacterineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Sorangiineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Roseiflexineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Nannocystineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "PVC group,", "") %>%
  mutate_at("Named Lineage", str_replace, "FCB group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Cyanobacteria/Melainabacteria group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Klebsiella/Raoultella group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Chromobacterium group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Erythrobacter/Porphyrobacter group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Massilia group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Sinorhizobium/Ensifer group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Brucella/Ochrobactrum group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Rhizobium/Agrobacterium group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Azotobacter group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Alteromonas/Salinimonas group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Chryseobacterium group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Chlamydia/Chlamydophila group,", "") %>%
  mutate_at("Named Lineage", str_replace, "CUG-Ser1 clade,", "") %>%
  mutate_at("Named Lineage", str_replace, "Candida/Lodderomyces clade,", "") %>%
  mutate_at("Named Lineage", str_replace, "Agaricomycetes incertae sedis,", "") %>%
  mutate_at("Named Lineage", str_replace, "saccharomyceta,","") %>%
  mutate_at("Named Lineage", str_replace, "leotiomyceta,", "") %>%
  mutate_at("Named Lineage", str_replace, "Agaricomycetidae,", "") %>%
  mutate_at("Named Lineage", str_replace, "CUG-Ser2 clade,", "") %>%
  mutate_at("Named Lineage", str_replace, "dothideomyceta,", "") %>%
  mutate_at("Named Lineage", str_replace, "Pleosporomycetidae,", "") %>%
  mutate_at("Named Lineage", str_replace, "sordariomyceta,", "") %>%
  mutate_at("Named Lineage", str_replace, "Hypocreomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Acarosporomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Chaetothyriomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Dothideomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Eurotiomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Mycocaliciomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Phallomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Pleosporomycetidae incertae sedis,", "")%>%
  mutate_at("Named Lineage", str_replace, "Sordariomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Xylariomycetidae,", "")%>%
  mutate_at("Named Lineage", str_replace, "Phallomycetidae,", "")%>% 
  mutate_at("Named Lineage", str_replace, "Dothideomycetes incertae sedis,", "Dothideomycetes incertae sedis,Dothideomycetes incertae sedis,") %>%
  mutate_at("Named Lineage", str_replace, "Apansporoblastina,", "Apansporoblastina,Apansporoblastina,Apansporoblastina,") %>%
  mutate_at("Named Lineage", str_replace, "Microsporidia incertae sedis,", "Microsporidia incertae sedis,Microsporidia incertae sedis,Microsporidia incertae sedis,Microsporidia incertae sedis,") %>%
  mutate_at("Named Lineage", str_replace, "Tubulinosematoidea,", "Tubulinosematoidea,Tubulinosematoidea,Tubulinosematoidea,") %>%
  mutate_at("Named Lineage", str_replace, "Pleosporineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Taphrinomycotina incertae sedis,", "Taphrinomycotina incertae sedis,Taphrinomycotina incertae sedis, Taphrinomycotina incertae sedis,") %>%
  mutate_at("Named Lineage", str_replace, "Mucorineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Fungiina,", "") %>%
  mutate_at("Named Lineage", str_replace, "Suillineae,", "") %>%
  mutate_at("Named Lineage", str_replace, "Pansporoblastina,", "Pansporoblastina,Pansporoblastina,Pansporoblastina,")%>%
  mutate_at("Named Lineage", str_replace, "TACK group,", "") %>%
  mutate_at("Named Lineage", str_replace, "Stenosarchaea group,", "")%>%
  mutate_at("Named Lineage", str_replace, "Methanomada group,", "")%>%
  mutate_at("Named Lineage", str_replace, "DPANN group,", "")



# Split the "Named Lineage" column into separate taxonomic levels
tax2name.split <- tax2name %>%
  separate("Named Lineage", into = c("root","domain", "superkingdom", "phylum", "class", "order", "family", "genus", "species"), sep=",")

# Handle taxonomy data for fungi which has longer lineages
tax2name.split.fungi <- tax2name %>%
  separate("Named Lineage", into = c("root", "domain", "superkingdom", "clade", "kingdom", "subkingdom", "phylum", "subphylum", "class", "order", "family", "genus", "species"), sep=",")

# Import braken counts data 
otu_kraken <- vroom("data/braken_results/species_bracken.tsv")

# Extract the taxonomy information from the "otu_kraken" dataframe
tax <- as.data.frame(otu_kraken[1:2])

# Remove unnecessary columns from the "otu_kraken" dataframe
otu_kraken <- otu_kraken[-c(2,3, seq(5,length(otu_kraken),2))]

# Clean up taxonomy IDs 
tax <- tax %>%
  mutate(taxonomy_id = ifelse(taxonomy_id == "2715211", as.double("2758038"), as.double(taxonomy_id))) %>%
  mutate(taxonomy_id = ifelse(taxonomy_id == "2918672", as.double("2917790"), as.double(taxonomy_id)))

# Join the taxonomy information for bacteria/archaea
tax2 <- left_join(tax, tax2name.split, by=c("taxonomy_id" = "Taxid"))

# Remove unnecessary columns
tax2 <- tax2[-c(3, 4, 14)]
rownames(tax2) <- paste0("Species", 1:nrow(tax2))

# Clean up missing taxonomic levels 
tax2 <- tax2 %>%
  mutate(genus = ifelse(is.na(genus), ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family)), as.character(genus))) %>%
  mutate(family = ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family))) %>%
  mutate(order = ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class)),as.character(order))) %>%
  mutate(class = ifelse(is.na(class), ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum)),as.character(class))) %>%
  mutate(phylum = ifelse(is.na(phylum), as.character(superkingdom), as.character(phylum))) 

tax3 <- tax2[-c(1:3)]

# Join the taxonomy information for fungi
tax2_fungi <- left_join(tax, tax2name.split.fungi, by=c("taxonomy_id" = "Taxid"))

# Remove unnecessary columns 
tax2_fungi <- tax2_fungi[-c(3, 4, 5, 6, 7,8,10,12, 18)]
rownames(tax2_fungi) <- paste0("Species", 1:nrow(tax2_fungi))

# Clean up missing taxonomic levels 
tax2_fungi <- tax2_fungi %>%
  mutate(genus = ifelse(is.na(genus), ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(kingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family)), as.character(genus))) %>%
  mutate(family = ifelse(is.na(family), ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(kingdom), as.character(phylum)),as.character(class)),as.character(order)),  as.character(family))) %>%
  mutate(order = ifelse(is.na(order), ifelse(is.na(class), ifelse(is.na(phylum), as.character(kingdom), as.character(phylum)),as.character(class)),as.character(order))) %>%
  mutate(class = ifelse(is.na(class), ifelse(is.na(phylum), as.character(kingdom), as.character(phylum)),as.character(class))) %>%
  mutate(phylum = ifelse(is.na(phylum), as.character(kingdom), as.character(phylum))) 

tax3_fungi <- tax2_fungi[-c(1:2)]


# Convert for phyloseq
otu_kraken <- as.data.frame(otu_kraken)
rownames(otu_kraken) <- paste0("Species", 1:nrow(otu_kraken))
otu_kraken <- otu_kraken[-c(1)]
otu_kraken_table <- otu_table(otu_kraken, taxa_are_rows = TRUE)

# Create taxonomy objects for bacteria/archaea and fungi
tax_table_kraken <- tax_table(as.matrix(tax3))
tax_table_kraken_fungi <- tax_table(as.matrix(tax3_fungi))

# Load metadata 
metadata <- vroom("data/metadata_crabs.csv")
metadata_fix <- metadata %>%
  mutate(FilePrefix = ifelse(FilePrefix == "BLB_WG4_S19", "BLB_WG4", ifelse(FilePrefix == "BLR_WG4_S20", "BLR_WG4", as.character(FilePrefix))))
meta <- sample_data(metadata_fix)
rownames(meta) <- paste0(meta$FilePrefix, ".S.bracken_num")

# Create phyloseq objects for bacteria/archaea and fungi
physeq_krak <- phyloseq(otu_kraken_table, tax_table_kraken, meta)
physeq_krak_fungitax <- phyloseq(otu_kraken_table, tax_table_kraken_fungi, meta)


physeq_krak
physeq_krak_fungitax

```


```{r nc}

# Summarize RPKM (Reads Per Kilobase Million) for each sample 
sampleSums(physeq_coverM)

# Calculate Kracken2/Braken counts per sample
sampleSums(physeq_krak)

# Subset set to only bacteria
physeq_krak_bac <- subset_taxa(physeq_krak, superkingdom == "Bacteria")
 
# Subset set to only archaea
physeq_krak_arc <- subset_taxa(physeq_krak, superkingdom == "Archaea")

# Subset set to only fungi
physeq_krak_fun <- subset_taxa(physeq_krak_fungitax, kingdom == "Fungi")

# Calculate remaining counts per sample after subsetting for each of the three groups
sampleSums(physeq_krak_bac)
sampleSums(physeq_krak_arc)
sampleSums(physeq_krak_fun)

```


### Use decontam to assess contaminants
```{r decontam,  fig.height=4,fig.width=6}

## MAGS ##
# Run decontamination analysis using a threshold of 0.5 
contams.mag <- run_decontam_threshold(physeq_coverM, 0.5, "NEG_S33")

# Calculate the number of contaminants found by decontam
length(contams.mag[[1]])

# Plot decontam results
contams.mag[[2]]

## Kraken - Bacteria ##
# Run decontamination analysis using a threshold of 0.5 
contams.bact <- run_decontam_threshold(physeq_krak_bac, 0.5, "NEG_S33")

# Calculate the number of contaminants found by decontam
length(contams.bact[[1]])

# Plot decontam results
contams.bact[[2]]

## Kraken - Archaea ##
# Run decontamination analysis using a threshold of 0.5 
contams.arc<- run_decontam_threshold(physeq_krak_arc, 0.5, "NEG_S33")

# Calculate the number of contaminants found by decontam
length(contams.arc[[1]])

# Plot decontam results
contams.arc[[2]]

## Kraken - Fungi ##
# Run decontamination analysis using a threshold of 0.5 
contams.fungi <- run_decontam_threshold(physeq_krak_fun, 0.5, "NEG_S33")

# Calculate the number of contaminants found by decontam
length(contams.fungi[[1]])

# Plot decontam results
contams.fungi[[2]]

```

###  Remove contaminants and negative control
```{r remove}

## MAG ## 
# Obtain a list of all taxa and making a new list of all taxa except those that are contaminants
all_taxa.mag <- taxa_names(physeq_coverM)
newtaxa.mag <- all_taxa.mag[!(all_taxa.mag %in% contams.mag[[1]])]

# Create a new phyloseq object by pruning the taxa identified as contaminants
physeq_coverM_nC <- prune_taxa(newtaxa.mag, physeq_coverM)

# Remove negative control (NEG_S33) 
physeq_coverM_nC <- subset_samples(physeq_coverM_nC, FilePrefix != "NEG_S33")

## Kraken - Bacteria ##
# Obtain a list of all taxa and making a new list of all taxa except those that are contaminants
all_taxa.bac <- taxa_names(physeq_krak_bac)
newtaxa.bac <- all_taxa.bac[!(all_taxa.bac %in% contams.bact[[1]])]

# Create a new phyloseq object by pruning the taxa identified as contaminants
physeq_krak_nC_bac <- prune_taxa(newtaxa.bac, physeq_krak_bac)

# Remove negative control (NEG_S33) 
physeq_krak_nC_bac <- subset_samples(physeq_krak_nC_bac, FilePrefix != "NEG_S33")

## Kraken - Archaea ##
# Obtain a list of all taxa and making a new list of all taxa except those that are contaminants
all_taxa.arc <- taxa_names(physeq_krak_arc)
newtaxa.arc <- all_taxa.arc[!(all_taxa.arc %in% contams.arc[[1]])]

# Create a new phyloseq object by pruning the taxa identified as contaminants
physeq_krak_nC_arc <- prune_taxa(newtaxa.arc, physeq_krak_arc)

# Remove negative control (NEG_S33) 
physeq_krak_nC_arc <- subset_samples(physeq_krak_nC_arc, FilePrefix != "NEG_S33")

## Kraken - Fungi ##
# Obtain a list of all taxa and making a new list of all taxa except those that are contaminants
all_taxa.fun <- taxa_names(physeq_krak_fun)
newtaxa.fun <- all_taxa.fun[!(all_taxa.fun %in% contams.fungi[[1]])]

# Create a new phyloseq object by pruning the taxa identified as contaminants
physeq_krak_nC_fun <- prune_taxa(newtaxa.fun, physeq_krak_fun)

physeq_krak_nC_fun <- subset_samples(physeq_krak_nC_fun, FilePrefix != "NEG_S33")
```


# Ordinations / Beta-Diversity

### DIET ###
###  Is there a relationship between diet and bacterial community composition in the guts of crabs?
###  Do herbivorous crabs have different microbial community members than detritivores? 

### GRADE ###
###  Is there a relationship between terrestrial grades and microbial community composition in the guts of crabs?

### PHYLOGENY ###
###  Is there a taxonomic signal (in the crabs) when examining microbial community composition?

```{r fact}

## MAG ##

# Reorder the factor levels fo 'Diet'
physeq_coverM_nC@sam_data$Diet <- factor(physeq_coverM_nC@sam_data$Diet,
                                         levels = c("Detritivore/filter feeder", "Detritivore/scavenger",
                                                    "Herbivore", "Omnivore", "Carnivore" ))

# Reorder the factor levels for 'Diet_group'
physeq_coverM_nC@sam_data$Diet_group <- factor(physeq_coverM_nC@sam_data$Diet_group,
                                         levels = c("Detritivore", "Herbivore", "Carnivore/Omnivore"))

# Reorder the factor levels for 'Terrestrial.Grade'
physeq_coverM_nC@sam_data$Terrestrial.Grade <- factor(physeq_coverM_nC@sam_data$Terrestrial.Grade,
                                         levels = c("TP:1a, Grade I", "TP:1a, Grade II", "TP:1b, Grade III",
                                                    "TP:1b, Grade IV", "TP:1a, Grade V" , "TP:1b, Grade V"))

# Reorder the factor levels for 'Grouped_Grade'
physeq_coverM_nC@sam_data$Grouped_Grade <- factor(physeq_coverM_nC@sam_data$Grouped_Grade,
                                         levels = c("low (I-II)", "mid (III)", "high (IV-V)"))

# Reorder the factor levels for 'Genus'
physeq_coverM_nC@sam_data$Genus <- factor(physeq_coverM_nC@sam_data$Genus,
                                         levels = c("Petrolisthes", "Birgus", "Tuerkayana",
                                                    "Gecarcoidea", "Geograpsus" , "Ocypode"))

# Reorder the factor levels for 'Species'
physeq_coverM_nC@sam_data$Species <- factor(physeq_coverM_nC@sam_data$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))


## KRAKEN - Bacteria ## 
# Reorder the factor levels as above
physeq_krak_nC_bac@sam_data$Diet <- factor(physeq_krak_nC_bac@sam_data$Diet,
                                         levels = c("Detritivore/filter feeder", "Detritivore/scavenger",
                                                    "Herbivore", "Omnivore", "Carnivore" ))

physeq_krak_nC_bac@sam_data$Diet_group <- factor(physeq_krak_nC_bac@sam_data$Diet_group,
                                         levels = c("Detritivore", "Herbivore", "Carnivore/Omnivore"))

physeq_krak_nC_bac@sam_data$Terrestrial.Grade <- factor(physeq_krak_nC_bac@sam_data$Terrestrial.Grade,
                                         levels = c("TP:1a, Grade I", "TP:1a, Grade II", "TP:1b, Grade III",
                                                    "TP:1b, Grade IV", "TP:1a, Grade V" , "TP:1b, Grade V"))

physeq_krak_nC_bac@sam_data$Grouped_Grade <- factor(physeq_krak_nC_bac@sam_data$Grouped_Grade,
                                         levels = c("low (I-II)", "mid (III)", "high (IV-V)"))

physeq_krak_nC_bac@sam_data$Genus <- factor(physeq_krak_nC_bac@sam_data$Genus,
                                         levels = c("Petrolisthes", "Birgus", "Tuerkayana",
                                                    "Gecarcoidea", "Geograpsus" , "Ocypode"))

physeq_krak_nC_bac@sam_data$Species <- factor(physeq_krak_nC_bac@sam_data$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))


## KRAKEN - Archaea ## 
# Reorder the factor levels as above
physeq_krak_nC_arc@sam_data$Diet <- factor(physeq_krak_nC_arc@sam_data$Diet,
                                         levels = c("Detritivore/filter feeder", "Detritivore/scavenger",
                                                    "Herbivore", "Omnivore", "Carnivore" ))

physeq_krak_nC_arc@sam_data$Diet_group <- factor(physeq_krak_nC_arc@sam_data$Diet_group,
                                         levels = c("Detritivore", "Herbivore", "Carnivore/Omnivore"))

physeq_krak_nC_arc@sam_data$Terrestrial.Grade <- factor(physeq_krak_nC_arc@sam_data$Terrestrial.Grade,
                                         levels = c("TP:1a, Grade I", "TP:1a, Grade II", "TP:1b, Grade III",
                                                    "TP:1b, Grade IV", "TP:1a, Grade V" , "TP:1b, Grade V"))
physeq_krak_nC_arc@sam_data$Grouped_Grade <- factor(physeq_krak_nC_arc@sam_data$Grouped_Grade,
                                         levels = c("low (I-II)", "mid (III)", "high (IV-V)"))

physeq_krak_nC_arc@sam_data$Genus <- factor(physeq_krak_nC_arc@sam_data$Genus,
                                         levels = c("Petrolisthes", "Birgus", "Tuerkayana",
                                                    "Gecarcoidea", "Geograpsus" , "Ocypode"))

physeq_krak_nC_arc@sam_data$Species <- factor(physeq_krak_nC_arc@sam_data$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))


## KRAKEN - Fungi ## 
# Reorder the factor levels as above
physeq_krak_nC_fun@sam_data$Diet <- factor(physeq_krak_nC_fun@sam_data$Diet,
                                         levels = c("Detritivore/filter feeder", "Detritivore/scavenger",
                                                    "Herbivore", "Omnivore", "Carnivore" ))

physeq_krak_nC_fun@sam_data$Diet_group <- factor(physeq_krak_nC_fun@sam_data$Diet_group,
                                         levels = c("Detritivore", "Herbivore", "Carnivore/Omnivore"))

physeq_krak_nC_fun@sam_data$Terrestrial.Grade <- factor(physeq_krak_nC_fun@sam_data$Terrestrial.Grade,
                                         levels = c("TP:1a, Grade I", "TP:1a, Grade II", "TP:1b, Grade III",
                                                    "TP:1b, Grade IV", "TP:1a, Grade V" , "TP:1b, Grade V"))

physeq_krak_nC_fun@sam_data$Grouped_Grade <- factor(physeq_krak_nC_fun@sam_data$Grouped_Grade,
                                         levels = c("low (I-II)", "mid (III)", "high (IV-V)"))

physeq_krak_nC_fun@sam_data$Genus <- factor(physeq_krak_nC_fun@sam_data$Genus,
                                         levels = c("Petrolisthes", "Birgus", "Tuerkayana",
                                                    "Gecarcoidea", "Geograpsus" , "Ocypode"))

physeq_krak_nC_fun@sam_data$Species <- factor(physeq_krak_nC_fun@sam_data$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))


```

```{r ordination, eval=FALSE, include=FALSE}

# MAG

# Bray-Curtis transformation of counts for the MAG data
rpkm_bray <- ordinate(
  physeq = physeq_coverM_nC, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet" 
rpkm_bray_crab_diet = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Diet",
  color = "Diet") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

# PCoA plot colored and shaped by "Terrestrial Grade" 
rpkm_bray_crab_grade = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Terrestrial.Grade",
  color = "Terrestrial.Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
rpkm_bray_crab_phylo = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

# KRAKEN - Bacteria

# Bray-Curtis transformation of counts for the Kraken2 data
krak_bray <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet" 
krak_bray_crab_diet = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Diet",
  color = "Diet") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

# PCoA plot colored and shaped by "Terrestrial.Grade" 
krak_bray_crab_grade = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Terrestrial.Grade",
  color = "Terrestrial.Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25)) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
krak_bray_crab_phylo = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


# All plots
(rpkm_bray_crab_diet + rpkm_bray_crab_grade + rpkm_bray_crab_phylo )/ (krak_bray_crab_diet + krak_bray_crab_grade +krak_bray_crab_phylo )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')

# Save as PNG
ggsave("plots/ordination/mag_kraken_ordinations.png", dpi=300, device = "png", height = 6, width= 10, units = "in")


```

```{r ordination2, fig.height=6,fig.width=10}

#MAG

# Bray-Curtis transformation of counts for the MAG data
rpkm_bray <- ordinate(
  physeq = physeq_coverM_nC, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet_group" 
rpkm_bray_crab_diet = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Diet_group",
  color = "Diet_group") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))

# PCoA plot colored and shaped by "Grade" 
rpkm_bray_crab_grade = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Grade",
  color = "Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
rpkm_bray_crab_phylo = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

# Kraken

# Bray-Curtis transformation of counts for the Kraken2 bacteria data
krak_bray <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet_group" 
krak_bray_crab_diet = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Diet_group",
  color = "Diet_group") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))

# PCoA plot colored and shaped by "Grade" 
krak_bray_crab_grade = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Grade",
  color = "Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25)) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
krak_bray_crab_phylo = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


# All plots
(rpkm_bray_crab_diet + rpkm_bray_crab_grade + rpkm_bray_crab_phylo )/ (krak_bray_crab_diet + krak_bray_crab_grade +krak_bray_crab_phylo )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')

# Save as PNG
ggsave("plots/ordination/mag_kraken_ordinations_diffgroup.png", dpi=300, device = "png", height = 6, width= 10, units = "in")


```

```{r ordination3, eval=FALSE, include=FALSE}

#MAG

#Bray-Curtis transformation of counts
rpkm_bray <- ordinate(
  physeq = physeq_coverM_nC, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
rpkm_bray_crab_diet = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Diet_herbivore",
  color = "Diet_herbivore") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


rpkm_bray_crab_grade = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Grouped_Grade",
  color = "Grouped_Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

rpkm_bray_crab_phylo = plot_ordination(
  physeq = physeq_coverM_nC,
  ordination = rpkm_bray,
  shape = "Infraorder",
  color = "Infraorder") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*rpkm_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*rpkm_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

#Kraken

#Bray-Curtis transformation of counts
krak_bray <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Diet_herbivore",
  color = "Diet_herbivore") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


krak_bray_crab_grade = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Grouped_Grade",
  color = "Grouped_Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray,
  shape = "Infraorder",
  color = "Infraorder") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


#all plots
(rpkm_bray_crab_diet + rpkm_bray_crab_grade + rpkm_bray_crab_phylo )/ (krak_bray_crab_diet + krak_bray_crab_grade +krak_bray_crab_phylo )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')


ggsave("plots/ordination/mag_kraken_ordinations_diffgroup2.png", dpi=300, device = "png", height = 6, width= 10, units = "in")


```

```{r krak_split, eval=FALSE, include=FALSE}

#Bacteria 
#Bray-Curtis transformation of counts
krak_bray_bac <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Diet",
  color = "Diet") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

krak_bray_crab_grade_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Terrestrial.Grade",
  color = "Terrestrial.Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

#Archaea 
#Bray-Curtis transformation of counts
krak_bray_arc <- ordinate(
  physeq = physeq_krak_nC_arc, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Diet",
  color = "Diet") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

krak_bray_crab_grade_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Terrestrial.Grade",
  color = "Terrestrial.Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


#Fungi 
#Bray-Curtis transformation of counts
krak_bray_fun <- ordinate(
  physeq = physeq_krak_nC_fun, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Diet",
  color = "Diet") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

krak_bray_crab_grade_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Terrestrial.Grade",
  color = "Terrestrial.Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )



#all plots
(krak_bray_crab_diet_bac + krak_bray_crab_grade_bac +krak_bray_crab_phylo_bac ) / (krak_bray_crab_diet_arc + krak_bray_crab_grade_arc +krak_bray_crab_phylo_arc ) / (krak_bray_crab_diet_fun + krak_bray_crab_grade_fun +krak_bray_crab_phylo_fun )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')


ggsave("plots/ordination/split_kraken_ordinations.png", dpi=300, device = "png", height = 8, width= 10, units = "in")


```

```{r krak_split2, fig.height=8,fig.width=10}

# Bacteria 

# Bray-Curtis transformation of counts for the Kraken2 bacteria data
krak_bray_bac <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet_group" 
krak_bray_crab_diet_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Diet_group",
  color = "Diet_group") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


# PCoA plot colored and shaped by "Grade" 
krak_bray_crab_grade_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Grade",
  color = "Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25)) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
krak_bray_crab_phylo_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

# Archaea 
# Bray-Curtis transformation of counts for the Kraken2 archaea data
krak_bray_arc <- ordinate(
  physeq = physeq_krak_nC_arc, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet_group" 
krak_bray_crab_diet_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Diet_group",
  color = "Diet_group") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


# PCoA plot colored and shaped by "Grade" 
krak_bray_crab_grade_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Grade",
  color = "Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25)) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
krak_bray_crab_phylo_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


# Fungi 
# Bray-Curtis transformation of counts for the Kraken2 - fungi data
krak_bray_fun <- ordinate(
  physeq = physeq_krak_nC_fun, 
  method = "PCoA", 
  distance = "bray")

# PCoA plot colored and shaped by "Diet_group" 
krak_bray_crab_diet_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Diet_group",
  color = "Diet_group") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) ) + 
  guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))

# PCoA plot colored and shaped by "Grade" 
krak_bray_crab_grade_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Grade",
  color = "Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25)) + 
  guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

# PCoA plot colored and shaped by "Genus" 
krak_bray_crab_phylo_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Genus",
  color = "Genus") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance")) + 
  scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )



# All plots
(krak_bray_crab_diet_bac + krak_bray_crab_grade_bac +krak_bray_crab_phylo_bac ) / (krak_bray_crab_diet_arc + krak_bray_crab_grade_arc +krak_bray_crab_phylo_arc ) / (krak_bray_crab_diet_fun + krak_bray_crab_grade_fun +krak_bray_crab_phylo_fun )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')

# Save as PNG
ggsave("plots/ordination/split_kraken_ordinations_diffgroup.png", dpi=300, device = "png", height = 8, width= 10, units = "in")


```

```{r krak_split3, eval=FALSE, include=FALSE}

#Bacteria 
#Bray-Curtis transformation of counts
krak_bray_bac <- ordinate(
  physeq = physeq_krak_nC_bac, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Diet_herbivore",
  color = "Diet_herbivore") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


krak_bray_crab_grade_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Grouped_Grade",
  color = "Grouped_Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_bac = plot_ordination(
  physeq = physeq_krak_nC_bac,
  ordination = krak_bray_bac,
  shape = "Infraorder",
  color = "Infraorder") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_bac$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_bac$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )

#Archaea 
#Bray-Curtis transformation of counts
krak_bray_arc <- ordinate(
  physeq = physeq_krak_nC_arc, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Diet_herbivore",
  color = "Diet_herbivore") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


krak_bray_crab_grade_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Grouped_Grade",
  color = "Grouped_Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_arc = plot_ordination(
  physeq = physeq_krak_nC_arc,
  ordination = krak_bray_arc,
  shape = "Infraorder",
  color = "Infraorder") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_arc$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_arc$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )


#Fungi 
#Bray-Curtis transformation of counts
krak_bray_fun <- ordinate(
  physeq = physeq_krak_nC_fun, 
  method = "PCoA", 
  distance = "bray")

#BC transformation of counts
krak_bray_crab_diet_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Diet_herbivore",
  color = "Diet_herbivore") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "C", end=.7) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )+ guides(shape=guide_legend(title="Diet"), color=guide_legend(title="Diet"))


krak_bray_crab_grade_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Grouped_Grade",
  color = "Grouped_Grade") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "D", end=.8, direction=-1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25))+ guides(shape=guide_legend(title="Terrestrial Grade"), color=guide_legend(title="Terrestrial Grade"))

krak_bray_crab_phylo_fun = plot_ordination(
  physeq = physeq_krak_nC_fun,
  ordination = krak_bray_fun,
  shape = "Infraorder",
  color = "Infraorder") + 
  geom_point(size = 4, alpha=.7) + 
  theme(text = element_text(size=8))  + 
  scale_colour_viridis_d(option = "H", begin =.1) +
  xlab(paste0("PCoA 1: ",round(100*krak_bray_fun$values$Relative_eig[1], 1),"% variance")) +
  ylab(paste0("PCoA 2: ",round(100*krak_bray_fun$values$Relative_eig[2], 1),"% variance"))+ scale_shape_manual(values=c( 15, 16, 17, 18, 12, 25) )



#all plots
(krak_bray_crab_diet_bac + krak_bray_crab_grade_bac +krak_bray_crab_phylo_bac ) / (krak_bray_crab_diet_arc + krak_bray_crab_grade_arc +krak_bray_crab_phylo_arc ) / (krak_bray_crab_diet_fun + krak_bray_crab_grade_fun +krak_bray_crab_phylo_fun )+ plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')


ggsave("plots/ordination/split_kraken_ordinations_diffgroup2.png", dpi=300, device = "png", height = 8, width= 10, units = "in")


```

```{r adonis}
# Set the random seed for reproducibility
set.seed(5311)

# Create empty data frames to store the results
adonis_pvals <- data.frame()
pwadonis_pvals <- data.frame()
disp_pvals <- data.frame()

# Replication
# Group the samples by Diet_group, Grade, and Genus and get the count for each group
physeq_coverM_nC@sam_data %>% group_by(Diet_group, Grade, Genus) %>% tally()

# Calculate Bray-Curtis distances for different datasets
Dist.mag.bray <- phyloseq::distance(physeq_coverM_nC, method = "bray",
    type = "samples")

Dist.krak.bray_bac <- phyloseq::distance(physeq_krak_nC_bac, method = "bray",
    type = "samples")

Dist.krak.bray_arc <- phyloseq::distance(physeq_krak_nC_arc, method = "bray",
    type = "samples")

Dist.krak.bray_fun <- phyloseq::distance(physeq_krak_nC_fun, method = "bray",
    type = "samples")

# Perform adonis tests for different combinations of predictors (Grade, Diet_group, Genus, Family, Infraorder, Ocean) 
# To assess best overall model
# For each adonis test, calculate the p-values and other statistics

adon.res = adonis2(formula = Dist.mag.bray ~ Grade + Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 4)

# Combine the results of adonis tests into a data frame
adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Diet_group", "Infraorder", "Ocean"), 
                         Model = "Grade + Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

# Bind the results of this adonis test to the overall results
adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

# Similar adonis tests are performed for different predictor combinations below
# The results are collected and appended to the adonis_pvals data frame

adon.res = adonis2(formula = Dist.mag.bray ~ Grade + Genus + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Grade + Family + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Family", "Ocean"), 
                         Model = "Grade + Family + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Grade + Infraorder + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Infraorder", "Ocean"), 
                         Model = "Grade + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Diet_group", "Infraorder", "Ocean"), 
                         Model = "Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Genus + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 2)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Genus", "Ocean"), 
                         Model = "Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Grade + Diet_group + Genus + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Diet_group", "Genus", "Ocean"), 
                         Model = "Grade + Diet_Group + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~ Grade +  Genus + Diet_group +Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #grade/genus

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.mag.bray ~  Diet_group + Genus + Grade+ Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Diet_group", "Genus", "Grade", "Ocean"), 
                         Model = "Diet_group + Genus + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.mag.bray ~  Diet_group +  Grade+ Genus+ Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Diet_group", "Grade", "Genus", "Ocean"), 
                         Model = "Diet_group + Grade + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.mag.bray ~  Genus + Grade + Diet_group + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Genus", "Grade", "Ocean"), 
                         Model = "Genus + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.mag.bray ~  Family + Grade + Diet_group + Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Family", "Grade", "Ocean"), 
                         Model = "Family + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.mag.bray ~  Genus + Diet_group + Grade+ Ocean, data = as(sample_data(physeq_coverM_nC), 'data.frame'), permutations = 9999)#grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "MAG", 
                         Category = c("Genus",  "Grade", "Ocean"), 
                         Model = "Genus + Diet_group + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

# Perform pairwise adonis tests to compare groups within each predictor category (e.g., pairwise comparisons of Grade, Diet_group, Genus, etc.)
# Calculate the p-values for these pairwise tests and store the results in the pwadonis_pvals data frame
# The results are collected and appended to the pwadonis_pvals data frame

pw.res = pairwise.adonis(Dist.mag.bray, as.factor(as(sample_data(physeq_coverM_nC), 'data.frame')[,"Grade"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "MAG"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)


pw.res = pairwise.adonis(Dist.mag.bray, as.factor(as(sample_data(physeq_coverM_nC), 'data.frame')[,"Diet_group"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "MAG"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.mag.bray, as.factor(as(sample_data(physeq_coverM_nC), 'data.frame')[,"Genus"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "MAG"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.mag.bray, as.factor(as(sample_data(physeq_coverM_nC), 'data.frame')[,"Family"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "MAG"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)


# Perform dispersion analysis for the different categories (e.g. Genus)
disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Genus"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

# Get p-values from the dispersion analysis
disp_dist.pvals <- get_disp_pvals(disp_dist.test)

# Create a data frame to store the results 
disp.df <- data.frame(Data = "MAG", 
                      Category = "Genus", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

# Combine the results with the overall results
disp_pvals <- bind_rows(disp.df, disp_pvals)

# Perform post hoc Tukey's Honest Significant Difference (HSD) test to identify significant pairwise differences in dispersion
# The results are saved to a CSV file
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/mag_tukey_disp_genus.csv")

# Similar dispersion analyses are repeated below for other predictor categories (Family, Diet_group, Grade, Ocean, Infraorder)

disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Family"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "MAG", 
                      Category = "Family", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

# Perform post hoc Tukey's HSD test
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/mag_tukey_disp_family.csv")


disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Diet_group"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "MAG", 
                      Category = "Diet_group", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


# Perform post hoc Tukey's HSD test
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/mag_tukey_disp_diet.csv")


disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Grade"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "MAG", 
                      Category = "Grade", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

# Perform post hoc Tukey's HSD test
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-", extra = "merge") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/mag_tukey_disp_grade.csv")


disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Ocean"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "MAG", 
                      Category = "Ocean", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


# Perform post hoc Tukey's HSD test
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    mutate(pairs = gsub("Indo-Pacific", "IndoPacific", pairs)) %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/mag_tukey_disp_ocean.csv")


disp_dist <- betadisper(Dist.mag.bray, as(sample_data(physeq_coverM_nC),"data.frame")[, "Infraorder"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "MAG", 
                      Category = "Infraorder", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)




# Repeat statistical analysis above now for Kraken2 bacteria dataset 

adon.res = adonis2(formula = Dist.krak.bray_bac ~ Grade + Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Diet_group", "Infraorder", "Ocean"), 
                         Model = "Grade + Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Grade + Genus + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)



adon.res = adonis2(formula = Dist.krak.bray_bac ~ Family + Genus + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Family", "Ocean"), 
                         Model = "Grade + Family + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Grade + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Infraorder", "Ocean"), 
                         Model = "Grade + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Diet_group", "Infraorder", "Ocean"), 
                         Model = "Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Genus + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 2)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Genus", "Ocean"), 
                         Model = "Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Grade + Diet_group + Genus + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Diet_group", "Genus", "Ocean"), 
                         Model = "Grade + Diet_Group + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~ Grade +  Genus + Diet_group +Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #grade/genus

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~  Diet_group + Genus + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Diet_group", "Genus", "Grade", "Ocean"), 
                         Model = "Diet_group + Genus + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_bac ~  Diet_group +  Grade+ Genus+ Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Diet_group", "Grade", "Genus", "Ocean"), 
                         Model = "Diet_group + Grade + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_bac ~  Genus + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Genus", "Grade", "Ocean"), 
                         Model = "Genus + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_bac ~  Family + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Family", "Grade", "Ocean"), 
                         Model = "Family + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_bac ~  Genus + Diet_group + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_bac), 'data.frame'), permutations = 9999)#grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Bacteria", 
                         Category = c("Genus",  "Grade", "Ocean"), 
                         Model = "Genus + Diet_group + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

pw.res = pairwise.adonis(Dist.krak.bray_bac, as.factor(as(sample_data(physeq_krak_nC_bac), 'data.frame')[,"Grade"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Bacteria"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)


pw.res = pairwise.adonis(Dist.krak.bray_bac, as.factor(as(sample_data(physeq_krak_nC_bac), 'data.frame')[,"Diet_group"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Bacteria"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_bac, as.factor(as(sample_data(physeq_krak_nC_bac), 'data.frame')[,"Genus"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Bacteria"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_bac, as.factor(as(sample_data(physeq_krak_nC_bac), 'data.frame')[,"Family"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Bacteria"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)


# Dispersion tests
disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Genus"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Bacteria", 
                      Category = "Genus", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_bac_tukey_disp_genus.csv")

disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Family"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Bacteria", 
                      Category = "Family", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_bac_tukey_disp_family.csv")

disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Diet_group"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Bacteria", 
                      Category = "Diet_group", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Grade"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Bacteria", 
                      Category = "Grade", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_bac_tukey_disp_grade.csv")

disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Ocean"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Bacteria", 
                      Category = "Ocean", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    mutate(pairs = gsub("Indo-Pacific", "IndoPacific", pairs)) %>%
    separate(pairs, c("pair1", "pair2"), sep = "-") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_bac_tukey_disp_ocean.csv")


disp_dist <- betadisper(Dist.krak.bray_bac, as(sample_data(physeq_krak_nC_bac),"data.frame")[, "Infraorder"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Infraorder", 
                      Category = "Genus", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

# Repeat statistical analysis above now for Kraken2 archaea dataset 

adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade + Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Diet_group", "Infraorder", "Ocean"), 
                         Model = "Grade + Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade + Genus + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade + Family + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Family", "Ocean"), 
                         Model = "Grade + Family + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Infraorder", "Ocean"), 
                         Model = "Grade + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Diet_group", "Infraorder", "Ocean"), 
                         Model = "Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Genus + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 2)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Genus", "Ocean"), 
                         Model = "Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade + Diet_group + Genus + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Diet_group", "Genus", "Ocean"), 
                         Model = "Grade + Diet_Group + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~ Grade +  Genus + Diet_group +Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #grade/genus

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~  Diet_group + Genus + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Diet_group", "Genus", "Grade", "Ocean"), 
                         Model = "Diet_group + Genus + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_arc ~  Diet_group +  Grade+ Genus+ Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Diet_group", "Grade", "Genus", "Ocean"), 
                         Model = "Diet_group + Grade + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_arc ~  Genus + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Genus", "Grade", "Ocean"), 
                         Model = "Genus + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_arc ~  Family + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Family", "Grade", "Ocean"), 
                         Model = "Family + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_arc ~  Genus + Diet_group + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_arc), 'data.frame'), permutations = 9999)#grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Archaea", 
                         Category = c("Genus",  "Grade", "Ocean"), 
                         Model = "Genus + Diet_group + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

pw.res = pairwise.adonis(Dist.krak.bray_arc, as.factor(as(sample_data(physeq_krak_nC_arc), 'data.frame')[,"Grade"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Archaea"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)


pw.res = pairwise.adonis(Dist.krak.bray_arc, as.factor(as(sample_data(physeq_krak_nC_arc), 'data.frame')[,"Diet_group"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Archaea"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_arc, as.factor(as(sample_data(physeq_krak_nC_arc), 'data.frame')[,"Genus"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Archaea"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_arc, as.factor(as(sample_data(physeq_krak_nC_arc), 'data.frame')[,"Family"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Archaea"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)

# Dispersion tests
disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Genus"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Genus", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Family"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Family", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Diet_group"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Diet_group", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Grade"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Grade", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Ocean"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Ocean", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


disp_dist <- betadisper(Dist.krak.bray_arc, as(sample_data(physeq_krak_nC_arc),"data.frame")[, "Infraorder"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Archaea", 
                      Category = "Infraorder", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

# Repeat statistical analysis above now for Kraken2 fungi dataset 

adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade + Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Diet_group", "Infraorder", "Ocean"), 
                         Model = "Grade + Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade + Genus + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade + Family + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include diet & genus in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Family", "Ocean"), 
                         Model = "Grade + Family + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Infraorder", "Ocean"), 
                         Model = "Grade + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Diet_group + Infraorder + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Diet_group", "Infraorder", "Ocean"), 
                         Model = "Diet_Group + Infraorder + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Genus + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999, by = 'margin') #can't include genus & diet in by=margin

adon.res.pval <- get_beta_pvals(adon.res, 2)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Genus", "Ocean"), 
                         Model = "Genus + Ocean",
                         Type= "margin",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade + Diet_group + Genus + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Diet_group", "Genus", "Ocean"), 
                         Model = "Grade + Diet_Group + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~ Grade +  Genus + Diet_group +Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #grade/genus

adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Grade", "Genus", "Ocean"), 
                         Model = "Grade + Genus + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~  Diet_group + Genus + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Diet_group", "Genus", "Grade", "Ocean"), 
                         Model = "Diet_group + Genus + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_fun ~  Diet_group +  Grade+ Genus+ Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #all

adon.res.pval <- get_beta_pvals(adon.res, 4)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Diet_group", "Grade", "Genus", "Ocean"), 
                         Model = "Diet_group + Grade + Genus + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_fun ~  Genus + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Genus", "Grade", "Ocean"), 
                         Model = "Genus + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

adon.res = adonis2(formula = Dist.krak.bray_fun ~  Family + Grade + Diet_group + Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999) #grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Family", "Grade", "Ocean"), 
                         Model = "Family + Grade + Diet_group + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)


adon.res = adonis2(formula = Dist.krak.bray_fun ~  Genus + Diet_group + Grade+ Ocean, data = as(sample_data(physeq_krak_nC_fun), 'data.frame'), permutations = 9999)#grade/genus


adon.res.pval <- get_beta_pvals(adon.res, 3)

adonis.df <- data.frame(Data = "Kraken - Fungi", 
                         Category = c("Genus",  "Grade", "Ocean"), 
                         Model = "Genus + Diet_group + Grade + Ocean",
                         Type= "sequential",
                         pval = unlist(adon.res.pval[[1]]),
                         Fmod = unlist(adon.res.pval[[2]]),
                         R2 = unlist(adon.res.pval[[3]]))

adonis_pvals <- bind_rows(adonis.df, adonis_pvals)

pw.res = pairwise.adonis(Dist.krak.bray_fun, as.factor(as(sample_data(physeq_krak_nC_fun), 'data.frame')[,"Grade"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Fungi"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)


pw.res = pairwise.adonis(Dist.krak.bray_fun, as.factor(as(sample_data(physeq_krak_nC_fun), 'data.frame')[,"Diet_group"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Fungi"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_fun, as.factor(as(sample_data(physeq_krak_nC_fun), 'data.frame')[,"Genus"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Fungi"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)
# pw.res.pretty <- pw.res %>%
#     separate(pairs, c("pair1", "pair2"), sep = " vs ") %>%
#     select(pair1, pair2, p.adjusted) %>%
#     pivot_wider(names_from = pair2, values_from = p.adjusted)

pw.res = pairwise.adonis(Dist.krak.bray_fun, as.factor(as(sample_data(physeq_krak_nC_fun), 'data.frame')[,"Family"]), perm = 9999, p.adjust.m = "BH")

pw.res$DataSet <- "Kraken - Fungi"

pwadonis_pvals <- bind_rows(pw.res, pwadonis_pvals)

# Disperion tests
disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Genus"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Genus", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

# post hoc test to see which pairs driving sig
tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-", extra = "merge") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_fun_tukey_disp_genus.csv")

disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Family"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Family", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

tukey.disp.pretty <- as_tibble(TukeyHSD(disp_dist)$group,
    rownames = "pairs") %>%
    separate(pairs, c("pair1", "pair2"), sep = "-", extra = "merge") %>%
    select(pair1, pair2, `p adj`) %>%
    pivot_wider(names_from = pair2, values_from = `p adj`)

#write.csv(tukey.disp.pretty, "results/posthoc_dispersion/krak_fun_tukey_disp_family.csv")


disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Diet_group"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Diet_group", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Grade"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Grade", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Ocean"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Ocean", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)

disp_dist <- betadisper(Dist.krak.bray_fun, as(sample_data(physeq_krak_nC_fun),"data.frame")[, "Infraorder"])
disp_dist.test <- permutest(disp_dist, permutations = 9999)

disp_dist.pvals <- get_disp_pvals(disp_dist.test)

disp.df <- data.frame(Data = "Kraken - Fungi", 
                      Category = "Infraorder", 
                      pval = disp_dist.pvals[[1]],
                      Fmod = disp_dist.pvals[[2]])

disp_pvals <- bind_rows(disp.df, disp_pvals)


# Correct p-values for adonis results using the BH method and add the corrected p-values to the data frame
adonis_pvals <- adonis_pvals %>%
    group_by(Data) %>%
    mutate(padj = p.adjust(pval, method = "BH"))

# Correct p-values for dispersion analysis results using the BH method and add the corrected p-values to the data frame
disp_pvals <- disp_pvals %>%
  group_by(Data) %>%
    mutate(padj = p.adjust(pval, method = "BH"))

# Write the adonis results to a CSV file
#write.csv(adonis_pvals, "results/adonis_pvals.csv", quote = FALSE, row.names = FALSE)

# Write the pairwise adonis results to a CSV file
#write.csv(pwadonis_pvals, "results/pwadonis_pvals.csv", quote = FALSE, row.names = FALSE)

# Write the dispersion analysis results to a CSV file
#write.csv(disp_pvals, "results/disp_pvals.csv", quote = FALSE, row.names = FALSE)



```


# Relative Abundance 
### DIET ###
### Is there a relationship between diet and bacterial community composition in the guts of crabs?
### Do herbivorous crabs have different microbial community members than detritivores? 

### GRADE ###
### Is there a relationship between terrestrial grades and microbial community composition in the guts of crabs?

### PHYLOGENY ###
### Is there a taxonomic signal (in the crabs) when examining microbial community composition?


```{r RA, eval=FALSE, include=FALSE}

# MAG 

# Transform counts to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

# Export from phylose to regular data frame 
df_ps.RA.filt <- psmelt(ps.RA.filt)

# Group data by various taxonomic levels and calculate mean Relative Abundance
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

# Write filtered and summarized data to a TSV file
write.table((avgs_grouped %>% filter(meanA > 5)), "results/mag_rpkm_RA_Diet_5perc.tsv", quote = FALSE,row.names = FALSE,sep = "\t")


# Group data by genus and prepare for plotting
avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Anaerorhabdus","Bacteroidales", "Breznakia", "Castellaniella","Citrobacter","Dysgonomonas","Enterococcus_B","Fimivivens","Kluyvera","Lachnospirales",  "Mangrovibacterium","Marinicella","Oleidesulfovibrio","Shewanella","UBA660",  "UJ101", "UPXZ01","Other" )))


# Create a bar plot showing mean Relative Abundance by Diet and genus
mag.diet.plot = ggplot(avgs_grouped_genus, aes(x = Diet, y = (meanA), fill = genus2)) +
  geom_bar(stat = "identity", position = "stack")  + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

mag.diet.plot


# Similar operations and plotting for Terrestrial Grade and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

write.table((avgs_grouped %>% filter(meanA > 5)), "results/mag_rpkm_RA_TG_5perc.tsv", quote = FALSE,row.names = FALSE,sep = "\t")

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Alphaproteobacteria","Anaerorhabdus","Bacteroidales","Bacteroides","Castellaniella"     ,"Citrobacter","Demequina","Dysgonomonas","Enterococcus_B","Enterococcus_C","Kluyvera","Macellibacteroides","Mangrovibacterium","Marinicella","Moheibacter","Oleidesulfovibrio","Paracoccus","Tenacibaculum","UBA10566","UJ101","UPXZ01","Vibrio","Other"  )))


mag.grade.plot = ggplot(avgs_grouped_genus, aes(x = Terrestrial.Grade, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

mag.grade.plot

# Similar operations and plotting for Host genus and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

write.table((avgs_grouped %>% filter(meanA > 5)), "results/mag_rpkm_RA_genus_5perc.tsv", quote = FALSE,row.names = FALSE,sep = "\t")


avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Alphaproteobacteria","Anaerorhabdus","Bacteroidales","Bacteroides","Breznakia","Castellaniella","Citrobacter","Demequina","Dysgonomonas","Enterococcus_B","Enterococcus_C","Fimivivens","Kluyvera","Lachnospirales","Mangrovibacterium","Marinicella","Moheibacter","Oleidesulfovibrio","Paracoccus","Parahaliea","Shewanella", "UBA660","UJ101","UPXZ01","Vibrio","Other" )))


mag.genus.plot = ggplot(avgs_grouped_genus, aes(x = Genus, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

mag.genus.plot

# Similar operations and plotting for Host species and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Aestuariibaculum","Alphaproteobacteria","Anaerorhabdus","Bacteroidales","Bacteroides","Breznakia","Castellaniella","Citrobacter","Demequina","Dysgonomonas","Enterococcus_B","Enterococcus_C","Fimivivens","Kluyvera","Lachnospirales","Macellibacteroides","Mangrovibacterium","Marinicella","Microbacterium","Moheibacter","Oleidesulfovibrio","Paracoccus","Parahaliea","Shewanella","Tenacibaculum","UBA10566","UBA660","UJ101","UPXZ01","Vibrio" ,"Other" )))


mag.species.plot = ggplot(avgs_grouped_genus, aes(x = Species, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

mag.species.plot

# Kraken - BACTERIA

# Transform counts to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

# Export to dataframe
df_ps.RA.filt <- psmelt(ps.RA)

# Group data by various taxonomic levels and calculate mean Relative Abundance
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

# Group data by genus and prepare for plotting
avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Bacillus","Chryseobacterium","Citrobacter","Clostridium","Enterobacter","Escherichia","Paracoccus","Pseudomonas","Shewanella","Streptomyces","Vibrio","uncultured bacterium","Other")))

# Create a bar plot showing mean Relative Abundance by Diet and genus
krak.diet.plot = ggplot(avgs_grouped_genus, aes(x = Diet, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

krak.diet.plot


# Similar operations and plotting for Terrestrial Grade and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Bacillus","Bacteroides","Castellaniella","Citrobacter","Clostridium","Enterobacter","Enterococcus","Escherichia","Flavobacterium","Microbacterium","Mycolicibacterium","Paenibacillus","Paracoccus","Pseudomonas","Shewanella","Streptomyces","Thiomicrorhabdus","Vibrio","uncultured bacterium","Other")))


krak.grade.plot = ggplot(avgs_grouped_genus, aes(x = Terrestrial.Grade, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

krak.grade.plot

# Similar operations and plotting for Host genus and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Acinetobacter","Bacillus","Bacteroides","Castellaniella","Chryseobacterium","Citrobacter","Clostridium","Enterobacter","Enterococcus","Escherichia","Flavobacterium","Microbacterium","Paracoccus","Pseudomonas","Shewanella","Staphylococcus","Streptomyces","Vibrio","uncultured bacterium","Other" )))


krak.genus.plot = ggplot(avgs_grouped_genus, aes(x = Genus, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

krak.genus.plot

# Similar operations and plotting for Host species and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 10, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Acinetobacter","Bacillus","Citrobacter","Clostridium","Enterobacter","Escherichia","Lactococcus","Mycolicibacterium","Paenibacillus","Paracoccus","Pseudomonas","Shewanella","Staphylococcus","Streptomyces", "Thiomicrorhabdus","Vibrio","uncultured bacterium","Other" )))


krak.species.plot = ggplot(avgs_grouped_genus, aes(x = Species, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

krak.species.plot

# Save plots as PNG files
mag.diet.plot / krak.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_diet.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

mag.grade.plot / krak.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_grade.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

mag.genus.plot / krak.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_genus.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

mag.species.plot / krak.species.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_species.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_arc_fun, eval=FALSE, include=FALSE}

# Kraken - Archaea

# Transform counts to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

# Export data to dataframe
df_ps.RA.filt <- psmelt(ps.RA)

# Group data by various taxonomic levels and calculate mean Relative Abundance
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

# Group data by genus and prepare for plotting
avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c( "Methanobrevibacter","Methanosarcina","unclassified Candidatus Methanoperedenaceae","uncultured Methanosarcinales archaeon","uncultured archaeon","Other")))

# Create a bar plot showing mean Relative Abundance by Diet and genus
arc.diet.plot = ggplot(avgs_grouped_genus, aes(x = Diet, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

arc.diet.plot

# Similar operations and plotting for Terrestrial Grade and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Halorubrum","Candidatus Lokiarchaeota archaeon","Methanobacterium","Methanobrevibacter","Methanococcus","Methanosarcina","Natrinema","Thermococcus","unclassified Candidatus Methanoperedenaceae", "uncultured Methanosarcinales archaeon","uncultured archaeon","uncultured euryarchaeote","Other")))


arc.grade.plot = ggplot(avgs_grouped_genus, aes(x = Terrestrial.Grade, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

arc.grade.plot

# Similar operations and plotting for Host genus and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Halorubrum","Candidatus Lokiarchaeota archaeon","Methanobacterium","Methanobrevibacter","Methanococcus","Methanosarcina","Natrinema","Thermococcus","unclassified Candidatus Methanoperedenaceae", "uncultured Methanosarcinales archaeon","uncultured archaeon","uncultured euryarchaeote","Other")))


arc.genus.plot = ggplot(avgs_grouped_genus, aes(x = Genus, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

arc.genus.plot

# Similar operations and plotting for Host species and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 10, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Halorubrum","Candidatus Lokiarchaeota archaeon","Methanobacterium","Methanobrevibacter","Methanococcus","Methanosarcina","Natrinema","Thermococcus","unclassified Candidatus Methanoperedenaceae", "uncultured Methanosarcinales archaeon","uncultured archaeon","uncultured euryarchaeote","Other")))


arc.species.plot = ggplot(avgs_grouped_genus, aes(x = Species, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

arc.species.plot


# Kraken - Fungi

# Transform counts to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

# Export to dataframe
df_ps.RA.filt <- psmelt(ps.RA)

# Group data by various taxonomic levels and calculate mean Relative Abundance
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

# Group data by genus and prepare for plotting
avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Aspergillus","Candida","Colletotrichum","Debaryomyces","Fusarium","Hirsutella","Kwoniella","Rhizophagus","Saccharomyces","Talaromyces","Trametes","Trichoderma","uncultured fungus","Other")))

# Create a bar plot showing mean Relative Abundance by Diet and genus
fun.diet.plot = ggplot(avgs_grouped_genus, aes(x = Diet, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

fun.diet.plot


# Similar operations and plotting for Terrestrial Grade and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Aspergillus","Botryosphaeriales","Candida","Colletotrichum","Debaryomyces","Epichloe","Fusarium","Kwoniella","Leptosphaeria","Moesziomyces","Penicillium","Pyrenophora","Rhizophagus","Saccharomyces","Saccharomycopsis","Talaromyces","Torulaspora","Trametes","Trichoderma","uncultured fungus","Ustilago","Other")))


fun.grade.plot = ggplot(avgs_grouped_genus, aes(x = Terrestrial.Grade, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

fun.grade.plot

# Similar operations and plotting for Host genus and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 5, "Other", as.character(genus)))%>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Aspergillus","Botryosphaeriales","Candida","Colletotrichum","Debaryomyces","Fusarium","Hirsutella","Kwoniella","Penicillium","Pyrenophora","Rhizophagus","Saccharomyces","Saccharomycopsis","Talaromyces","Torulaspora","Trametes","Trichoderma","uncultured fungus","Other")))


fun.genus.plot = ggplot(avgs_grouped_genus, aes(x = Genus, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack") + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

fun.genus.plot

# Similar operations and plotting for Host species and genus
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_genus <- avgs_grouped %>%
  group_by(genus) %>%
  mutate(genus2 = ifelse(sum(meanA) < 10, "Other", as.character(genus))) %>%
  mutate(genus2 = factor(genus2, 
                         levels = c("Aspergillus","Botryosphaeriales","Candida","Colletotrichum","Debaryomyces","Epichloe","Exophiala","Fusarium","Hirsutella","Kwoniella","Lentinula","Leptosphaeria","Moesziomyces","Penicillium","Pyrenophora","Rhizophagus","Saccharomyces","Saccharomycopsis","Talaromyces","Torulaspora","Trametes","Trichoderma","Ustilago","uncultured fungus","Other")))


fun.species.plot = ggplot(avgs_grouped_genus, aes(x = Species, y = (meanA), fill = genus2)) + 
  geom_bar(stat = "identity", position = "stack")  + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

fun.species.plot

# Save plots as PNG files
arc.diet.plot / fun.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_diet_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

arc.grade.plot / fun.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_grade_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

arc.genus.plot / fun.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_genus_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

arc.species.plot / fun.species.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_genus_species_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_order, eval=FALSE, include=FALSE}

#MAG 
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.order)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Actinomycetales","Bacteroidales","Burkholderiales","Desulfovibrionales","Enterobacterales","Erysipelotrichales", "Flavobacteriales","Lachnospirales","Lactobacillales","Oscillospirales", "RF39", "Rhodobacterales","Xanthomonadales","Other")))


plot = ggplot(avgs_grouped_order, aes(x = Diet, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.mag.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.mag.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Acidimicrobiales","Actinomycetales","Alphaproteobacteria","Bacteroidales","Burkholderiales","Desulfovibrionales","Enterobacterales","Erysipelotrichales","Flavobacteriales","Lachnospirales","Lactobacillales","Pseudomonadales","Rhizobiales","Rhodobacterales", "Xanthomonadales","Other"   )))


plot = ggplot(avgs_grouped_order, aes(x = Terrestrial.Grade, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.mag.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.mag.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Actinomycetales","Alphaproteobacteria","Bacteroidales","Burkholderiales","Desulfovibrionales","Enterobacterales","Erysipelotrichales","Flavobacteriales","Lachnospirales","Lactobacillales","Oscillospirales", "Pseudomonadales","RF39","Rhodobacterales","Xanthomonadales","Other")))


plot = ggplot(avgs_grouped_order, aes(x = Genus, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.mag.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.mag.genus.plot





#Kraken 
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.order)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Alteromonadales","Bacillales","Bacteroidales","Burkholderiales","Campylobacterales","Corynebacteriales","Cytophagales","Desulfovibrionales","Enterobacterales","Eubacteriales","Flavobacteriales","Hyphomicrobiales","Lactobacillales","Micrococcales","Moraxellales", "Mycoplasmatales","Pseudomonadales","Rhodobacterales","Streptomycetales","Vibrionales", "uncultured bacterium" ,"Other")))

plot = ggplot(avgs_grouped_order, aes(x = Diet, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.krak.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.krak.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Alteromonadales","Bacillales","Bacteroidales","Burkholderiales","Campylobacterales","Corynebacteriales","Cytophagales","Desulfovibrionales","Enterobacterales","Eubacteriales","Flavobacteriales","Hyphomicrobiales","Lactobacillales","Micrococcales","Moraxellales","Mycoplasmatales","Pseudomonadales","Rhodobacterales","Streptomycetales","Thiotrichales","Vibrionales","uncultured bacterium","Other")))



plot = ggplot(avgs_grouped_order, aes(x = Terrestrial.Grade, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.krak.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.krak.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Alteromonadales","Bacillales","Bacteroidales","Burkholderiales","Campylobacterales","Corynebacteriales","Cytophagales","Desulfovibrionales","Enterobacterales","Entomoplasmatales","Eubacteriales","Flavobacteriales","Hyphomicrobiales","Lactobacillales","Micrococcales","Moraxellales","Mycoplasmatales","Pseudomonadales","Rhodobacterales","Streptomycetales","uncultured bacterium","Vibrionales", "Other" )))


plot = ggplot(avgs_grouped_order, aes(x = Genus, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.krak.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.krak.genus.plot



o.mag.diet.plot / o.krak.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_diet.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

o.mag.grade.plot / o.krak.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_grade.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

o.mag.genus.plot / o.krak.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_genus.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_arc_fun_order, eval=FALSE, include=FALSE}
#Kraken - archaea
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.order)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Halobacteriales","Haloferacales","Methanobacteriales","Methanococcales","Methanosarcinales","Natrialbales","Sulfolobales","Thermococcales","uncultured archaeon","Other")))

plot = ggplot(avgs_grouped_order, aes(x = Diet, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.arc.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.arc.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Halobacteriales","Haloferacales","Methanobacteriales","Methanococcales","Methanomicrobiales","Methanosarcinales","Natrialbales","Nitrosopumilaceae","Nitrososphaerales", "Sulfolobales","Thermococcales","Thermoplasmatales","unclassified Lokiarchaeota", "uncultured archaeon","uncultured euryarchaeote","Other")))



plot = ggplot(avgs_grouped_order, aes(x = Terrestrial.Grade, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.arc.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.arc.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Halobacteriales","Haloferacales","Methanobacteriales","Methanococcales","Methanosarcinales","Natrialbales","Nitrosopumilaceae","Sulfolobales","Thermococcales","Thermoplasmatales","unclassified Lokiarchaeota","uncultured archaeon","Other")))


plot = ggplot(avgs_grouped_order, aes(x = Genus, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.arc.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.arc.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 10, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Desulfurococcales","Halobacteriales","Haloferacales","Methanobacteriales","Methanococcales","Methanosarcinales","Natrialbales","Nitrosopumilaceae","Sulfolobales","Thermococcales","Thermoplasmatales", "uncultured archaeon","uncultured euryarchaeote","Other")))


plot = ggplot(avgs_grouped_order, aes(x = Species, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.arc.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.arc.species.plot


#Kraken - Fungi
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.order)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c( "Agaricales","Chaetothyriales","Dothideomycetes incertae sedis", "Eurotiales","Glomerales","Glomerellales","Helotiales","Hypocreales","Mucorales", "Mycosphaerellales","Onygenales","Pleosporales","Polyporales","Saccharomycetales", "Sordariales","Tremellales","Trichosporonales", "Ustilaginales","uncultured fungus","Other")))

plot = ggplot(avgs_grouped_order, aes(x = Diet, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.fun.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.fun.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Agaricales","Boletales","Chaetothyriales", "Diaporthales","Dothideomycetes incertae sedis","Eurotiales","Glomerales","Glomerellales","Helotiales","Hypocreales","Mucorales","Mycosphaerellales","Onygenales","Pleosporales","Polyporales","Saccharomycetales","Sordariales", "Tremellales","Trichosporonales", "Ustilaginales","uncultured fungus","Other")))



plot = ggplot(avgs_grouped_order, aes(x = Terrestrial.Grade, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.fun.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.fun.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 5, "Other", as.character(order)))%>%
  mutate(order2 = factor(order2, 
                         levels = c("Agaricales","Boletales","Chaetothyriales","Diaporthales","Dothideomycetes incertae sedis","Eurotiales", "Glomerales","Glomerellales","Helotiales","Hypocreales","Mucorales","Mycosphaerellales","Onygenales","Pleosporales","Polyporales","Saccharomycetales","Sordariales","Tremellales","Trichosporonales","Ustilaginales","uncultured fungus","Other")))


plot = ggplot(avgs_grouped_order, aes(x = Genus, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.fun.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.fun.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_order <- avgs_grouped %>%
  group_by(order) %>%
  mutate(order2 = ifelse(sum(meanA) < 10, "Other", as.character(order))) %>%
  mutate(order2 = factor(order2, 
                         levels = c("Agaricales","Boletales","Chaetothyriales","Diaporthales","Dothideomycetes incertae sedis","Eurotiales","Glomerales","Glomerellales","Helotiales","Hypocreales","Mucorales","Mycosphaerellales", "Onygenales","Pleosporales","Polyporales","Saccharomycetales","Sordariales","Tremellales","Trichosporonales","Ustilaginales","uncultured fungus","Other" )))


plot = ggplot(avgs_grouped_order, aes(x = Species, y = (meanA), fill = order2)) + 
  geom_bar(stat = "identity", position = "stack") 

o.fun.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

o.fun.species.plot


o.arc.diet.plot / o.fun.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_diet_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

o.arc.grade.plot / o.fun.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_grade_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

o.arc.genus.plot / o.fun.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_genus_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

o.arc.species.plot / o.fun.species.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_order_species_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_class, eval=FALSE, include=FALSE}

#MAG 
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) 


plot = ggplot(avgs_grouped_class, aes(x = Diet, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.mag.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.mag.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class)))

plot = ggplot(avgs_grouped_class, aes(x = Terrestrial.Grade, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.mag.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.mag.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) 


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.mag.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.mag.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) 


plot = ggplot(avgs_grouped_class, aes(x = Species, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.mag.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.mag.species.plot




#Kraken 
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Gammaproteobacteria","Mollicutes","uncultured bacterium" ,"Other")))

plot = ggplot(avgs_grouped_class, aes(x = Diet, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.krak.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.krak.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Gammaproteobacteria","Mollicutes","uncultured bacterium" ,"Other")))



plot = ggplot(avgs_grouped_class, aes(x = Terrestrial.Grade, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.krak.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.krak.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                          levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Gammaproteobacteria","Mollicutes","uncultured bacterium" ,"Other")))



plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.krak.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.krak.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                          levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Fusobacteriia","Gammaproteobacteria","Mollicutes","Nostocales","Planctomycetia", "Sphingobacteriia","Spirochaetia","uncultured bacterium" ,"Other")))



plot = ggplot(avgs_grouped_class, aes(x = Species, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.krak.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.krak.species.plot


class.mag.diet.plot / class.krak.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/ra_class_diet.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.mag.grade.plot / class.krak.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/ra_class_grade.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.mag.genus.plot / class.krak.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/ra_class_genus.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.mag.species.plot / class.krak.species.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/ra_class_species.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_arc_fun_class, eval=FALSE, include=FALSE}
#Kraken - archaea
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Lokiarchaeota","Halobacteria","Methanobacteria", "Methanococci","Methanomicrobia","Thermococci", "Thermoplasmata","Thermoprotei","uncultured archaeon","environmental samples","Other")))

plot = ggplot(avgs_grouped_class, aes(x = Diet, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.arc.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.arc.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Heimdallarchaeota","Candidatus Lokiarchaeota","Halobacteria","Methanobacteria","Methanococci","Methanomicrobia","Nitrosopumilales","Nitrososphaeria","Thermococci","Thermoplasmata","Thermoprotei","uncultured archaeon",  "environmental samples","Other")))



plot = ggplot(avgs_grouped_class, aes(x = Terrestrial.Grade, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.arc.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.arc.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Lokiarchaeota","Halobacteria","Methanobacteria","Methanococci","Methanomicrobia","Nitrosopumilales","Thermococci","Thermoplasmata","Thermoprotei","uncultured archaeon","environmental samples","Other")))


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.arc.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.arc.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 10, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Lokiarchaeota","Halobacteria","Methanobacteria","Methanococci","Methanomicrobia","Nitrosopumilales" ,"Thermococci","Thermoplasmata","Thermoprotei","uncultured archaeon","environmental samples" ,"Other")))


plot = ggplot(avgs_grouped_class, aes(x = Species, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.arc.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.arc.species.plot


#Kraken - Fungi
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# diet 
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))

plot = ggplot(avgs_grouped_class, aes(x = Diet, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.fun.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.fun.diet.plot


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Terrestrial.Grade, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))



plot = ggplot(avgs_grouped_class, aes(x = Terrestrial.Grade, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.fun.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.fun.grade.plot

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class)))%>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.fun.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.fun.genus.plot

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 10, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))


plot = ggplot(avgs_grouped_class, aes(x = Species, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "stack") 

class.fun.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) 

class.fun.species.plot


class.arc.diet.plot / class.fun.diet.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_class_diet_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.arc.grade.plot / class.fun.grade.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_class_grade_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.arc.genus.plot / class.fun.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/ra_class_genus_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")

class.arc.species.plot / class.fun.species.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/rrelative_abundance/a_class_species_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_class_mag, eval=FALSE, include=FALSE}

#MAG 
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) 


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "dodge") 

split.class.mag.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~class2)

split.class.mag.genus.plot


#Kraken 
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                          levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Gammaproteobacteria","Mollicutes","uncultured bacterium" ,"Other")))



plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "dodge") 

split.class.krak.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~class2)

split.class.krak.genus.plot


split.class.mag.genus.plot / split.class.krak.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/split.ra_class_genus.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```

```{r RA_arc_fun_class_2, eval=FALSE, include=FALSE}
#Kraken - archaea
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Lokiarchaeota","Halobacteria","Methanobacteria","Methanococci","Methanomicrobia","Nitrosopumilales","Thermococci","Thermoplasmata","Thermoprotei","uncultured archaeon","environmental samples","Other")))


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "dodge") 

split.class.arc.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +facet_wrap(~class2)

split.class.arc.genus.plot


#Kraken - Fungi
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

df_ps.RA.filt <- psmelt(ps.RA.class)


#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class)))%>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))


plot = ggplot(avgs_grouped_class, aes(x = Genus, y = (meanA), fill = class2)) + 
  geom_bar(stat = "identity", position = "dodge") 

split.class.fun.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "class")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +facet_wrap(~class2)

split.class.fun.genus.plot



split.class.arc.genus.plot / split.class.fun.genus.plot + plot_annotation(tag_levels = 'A') + plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/split.ra_class_genus_af.png", dpi=300, device = "png", height = 12, width= 8, units = "in")


```


```{r RA_class_side, fig.height=8,fig.width=18}

# MAG 

# Transforming to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

# Aggregating data at the class level 
ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

# Converting the aggregated data to a data frame
df_ps.RA.filt <- psmelt(ps.RA.class)

# Creating a factor and italisizing taxonomic names
df_ps.RA.filt$Species <- factor(df_ps.RA.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"),
                                         labels = c("*Petrolisthes armatus*","*Petrolisthes bolivarensis*","*Petrolisthes donadio*","*Petrolisthes haigae*","*Petrolisthes nobilii*","*Petrolisthes* sp.","*Birgus latro*","*Tuerkayana celeste*" ,"*Tuerkayana magna*", "*Gecarcoidea humei*" , "*Gecarcoidea natalis*","*Geograpsus grayi*","*Ocypode ceratopthalmus*"))


# Calculating mean relative abundance for each crab species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Infraorder, Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# Grouping low-abundance classes into an "Other" category
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) 

# Assigning a label to identify the dataset as "MAG"
avgs_grouped_class$DataSet <- "MAG"

mag.avgs <- avgs_grouped_class

# Creating a stacked bar plot to visualize mean relative abundances of species grouped by class
side.class.mag.species.plot = ggplot(avgs_grouped_class, aes(x = (meanA), y = Species, fill = class2)) +
  geom_bar(stat = "identity", position = "stack", orientation = "y", width = 0.6) + 
  theme_bw() + 
  theme(text = element_text(size = 12)) + 
  xlab("Mean Relative Abundance") + 
  ylab("") + 
  guides(fill = guide_legend(title = "Class", nrow=4,byrow=TRUE)) + 
  scale_fill_viridis_d(option = "H") + 
  scale_y_discrete(limits = rev) + 
  facet_wrap(~Infraorder, scales = "free_y", strip.position="left", ncol=1) +
  theme(axis.text.y = element_markdown()) + 
  theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin())

# Displaying the final plot
side.class.mag.species.plot


# Kraken - Bacteria

# Transforming to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

# Aggregating data at the class level 
ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

# Converting the aggregated data to a data frame
df_ps.RA.filt <- psmelt(ps.RA.class)

# Creating a factor and italisizing taxonomic names
df_ps.RA.filt$Species <- factor(df_ps.RA.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"),
                                         labels = c("*Petrolisthes armatus*","*Petrolisthes bolivarensis*","*Petrolisthes donadio*","*Petrolisthes haigae*","*Petrolisthes nobilii*","*Petrolisthes* sp.","*Birgus latro*","*Tuerkayana celeste*" ,"*Tuerkayana magna*", "*Gecarcoidea humei*" , "*Gecarcoidea natalis*","*Geograpsus grayi*","*Ocypode ceratopthalmus*"))

# Calculating mean relative abundance for each crab species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Infraorder, Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# Grouping low-abundance classes into an "Other" category
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                          levels = c("Actinomycetia","Alphaproteobacteria","Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Epsilonproteobacteria","Flavobacteriia", "Fusobacteriia","Gammaproteobacteria","Mollicutes","Nostocales","Planctomycetia","Sphingobacteriia","Spirochaetia","uncultured bacterium" ,"Other")))

# Assigning a label to identify the dataset
avgs_grouped_class$DataSet <- "Kraken2 - Bacteria"

# Combining MAG and Kraken bacteria relative abundance data
both.avg <- full_join(mag.avgs, avgs_grouped_class)

# Creating a stacked bar plot to visualize mean relative abundances of species grouped by class
side.class.krak.species.plot = ggplot(avgs_grouped_class, aes(x = (meanA), y = Species, fill = class2)) + 
  geom_bar(stat = "identity", position = "stack", orientation = "y", width = 0.6) + 
  theme_bw() + 
  theme(text = element_text(size = 12)) +
  xlab("Mean Relative Abundance") + 
  ylab("") + 
  guides(fill = guide_legend(title = "Class", nrow=6,byrow=TRUE)) + 
  scale_fill_viridis_d(option = "H") + 
  scale_y_discrete(limits = rev) + 
  facet_wrap(~Infraorder, scales = "free_y", strip.position="left", ncol=1) +
  theme(axis.text.y = element_markdown()) + 
  theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin()) 

# Displaying the final plot
side.class.krak.species.plot

# Displaying a combined plot and saving as PNG
side.class.mag.species.plot + side.class.krak.species.plot + plot_annotation(tag_levels = 'A') #+ plot_layout(guides = 'collect')
ggsave("plots/relative_abundance/side.ra_class_genus.png", dpi=300, device = "png", height = 8, width= 18, units = "in")


# Repeating plotting with the dataset with MAG and Kraken data combined
both.avg  <- both.avg  %>%
  mutate(class2 = factor(class2, 
                          levels = c("Acidimicrobiia","Actinomycetia","Alphaproteobacteria",   "Bacilli","Bacteroidia","Betaproteobacteria","Clostridia","Cytophagia","Deltaproteobacteria","Desulfovibrionia", "Epsilonproteobacteria","Flavobacteriia","Fusobacteriia","Gammaproteobacteria","Mollicutes","Nostocales","Planctomycetia","Sphingobacteriia","Spirochaetia","Verrucomicrobiae","uncultured bacterium" , "Other")))

# Get a list of unique Order names across both datasets
fams_of_interest <- unique(both.avg$class2)

# Filter the data to include only classes of interest 
both.avg.filt <- both.avg  %>%
  mutate(class2 = ifelse(class %in% fams_of_interest[-c(1)], as.character(class), as.character(class2))) 

# Creating a stacked bar plot to visualize mean relative abundances of species grouped by class
side.class.mag.and.krak.species.plot = ggplot(both.avg, aes(x = (meanA), y = Species, fill = class2)) + 
  geom_bar(stat = "identity", position = "stack", orientation = "y", width = 0.6) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  xlab("Mean Relative Abundance") + 
  ylab("") + 
  guides(fill = guide_legend(title = "Class")) + 
  scale_fill_viridis_d(option = "H") + 
  scale_y_discrete(limits = rev) + 
  facet_wrap(~DataSet, scales = "free_y", ncol=2) +
  theme(axis.text.y = element_markdown()) + 
  theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin()) 

# Displaying and saving the final plot as a PNG
side.class.mag.and.krak.species.plot
ggsave("plots/relative_abundance/combo.ra_class_genus.png", dpi=300, device = "png", height = 8, width= 14, units = "in")


```


```{r RA_arc_fun_class_side, fig.height=8,fig.width=12}

# Kraken - Archaea

# Transforming to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

# Aggregating data at the class level 
ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

# Converting the aggregated data to a data frame
df_ps.RA.filt <- psmelt(ps.RA.class)

# Creating a factor and italisizing taxonomic names
df_ps.RA.filt$Species <- factor(df_ps.RA.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"),
                                         labels = c("*Petrolisthes armatus*","*Petrolisthes bolivarensis*","*Petrolisthes donadio*","*Petrolisthes haigae*","*Petrolisthes nobilii*","*Petrolisthes* sp.","*Birgus latro*","*Tuerkayana celeste*" ,"*Tuerkayana magna*", "*Gecarcoidea humei*" , "*Gecarcoidea natalis*","*Geograpsus grayi*","*Ocypode ceratopthalmus*"))

# Calculating mean relative abundance for each crab species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Infraorder, Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# Grouping low-abundance classes into an "Other" category
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class))) %>%
  mutate(class2 = factor(class2, 
                         levels = c("Candidatus Heimdallarchaeota", "Candidatus Lokiarchaeota","Halobacteria","Methanobacteria","Methanococci","Methanomicrobia","Nitrosopumilales","Nitrososphaeria","Thaumarchaeota incertae sedis" ,"Thermococci","Thermoplasmata","Thermoprotei","unclassified Candidatus Bathyarchaeota","uncultured archaeon","environmental samples","Other")))

# Creating a stacked bar plot to visualize mean relative abundances of species grouped by class
side.class.arc.genus.plot = ggplot(avgs_grouped_class, aes(x = (meanA), y = Species, fill = class2)) + 
  geom_bar(stat = "identity", position = "stack", orientation = "y", width = 0.6)  + 
  theme_bw() + 
  theme(text = element_text(size = 12)) +
  xlab("Mean Relative Abundance") + 
  ylab("") + 
  guides(fill = guide_legend(title = "Class")) + 
  scale_fill_viridis_d(option = "H") + 
  scale_y_discrete(limits = rev) + 
  facet_wrap(~Infraorder, scales = "free_y", strip.position="left", ncol=1) +
  theme(axis.text.y = element_markdown()) + 
  theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin()) 

# Displaying the plot and saving as a PNG
side.class.arc.genus.plot
ggsave("plots/relative_abundance/archaea/side.ra_class_genus_arc.png", dpi=300, device = "png", height = 8, width= 12, units = "in")


# Kraken - Fungi

# Transforming to Relative Abundance (RA)
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

# Aggregating data at the class level 
ps.RA.class <- tax_glom(ps.RA, taxrank = "class", NArm = FALSE)

# Converting the aggregated data to a data frame
df_ps.RA.filt <- psmelt(ps.RA.class)

# Creating a factor and italisizing taxonomic names
df_ps.RA.filt$Species <- factor(df_ps.RA.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"),
                                         labels = c("*Petrolisthes armatus*","*Petrolisthes bolivarensis*","*Petrolisthes donadio*","*Petrolisthes haigae*","*Petrolisthes nobilii*","*Petrolisthes* sp.","*Birgus latro*","*Tuerkayana celeste*" ,"*Tuerkayana magna*", "*Gecarcoidea humei*" , "*Gecarcoidea natalis*","*Geograpsus grayi*","*Ocypode ceratopthalmus*"))

# Calculating mean relative abundance for each crab species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Infraorder, Species, phylum, class) %>%
  summarise(meanA =  100 * mean(Abundance))

# Grouping low-abundance classes into an "Other" category
avgs_grouped_class <- avgs_grouped %>%
  group_by(class) %>%
  mutate(class2 = ifelse(sum(meanA) < 5, "Other", as.character(class)))%>%
  mutate(class2 = factor(class2, 
                         levels = c("Agaricomycetes","Dothideomycetes","Eurotiomycetes","Exobasidiomycetes","Glomeromycetes","Leotiomycetes","Mucoromycetes","Saccharomycetes",   "Sordariomycetes","Tremellomycetes","Ustilaginomycetes","uncultured fungus" , "Other")))

# Creating a stacked bar plot to visualize mean relative abundances of species grouped by class
side.class.fun.genus.plot = ggplot(avgs_grouped_class, aes(x = (meanA), y = Species, fill = class2)) + 
  geom_bar(stat = "identity", position = "stack", orientation = "y", width = 0.6)  + 
  theme_bw() + 
  theme(text = element_text(size = 12)) +
  xlab("Mean Relative Abundance") + 
  ylab("") + 
  guides(fill = guide_legend(title = "Class")) + 
  scale_fill_viridis_d(option = "H") + 
  scale_y_discrete(limits = rev) + 
  facet_wrap(~Infraorder, scales = "free_y", strip.position="left", ncol=1)+
  theme(axis.text.y = element_markdown()) + 
  theme(legend.position = "bottom", legend.box="vertical", legend.margin=margin()) 

# Displaying the plot and saving as a PNG
side.class.fun.genus.plot
ggsave("plots/relative_abundance/fungi/side.ra_class_genus_fungi.png", dpi=300, device = "png", height = 8, width= 10, units = "in")


#side.class.arc.genus.plot + side.class.fun.genus.plot + plot_annotation(tag_levels = 'A') #+ plot_layout(guides = 'collect')
#ggsave("plots/side.ra_class_genus_af.png", dpi=300, device = "png", height = 8, width= 18, units = "in")


```



```{r RA_split_variance, eval=FALSE, include=FALSE}

#MAG 
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) var(x) > 0.001, TRUE)

df_ps.RA.filt <- psmelt(ps.RA.gen)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


# diet 
plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

mag.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

mag.diet.plot

ggsave("plots/relative_abundance/mag/ra_genus_diet_mag_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

mag.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

mag.grade.plot

ggsave("plots/relative_abundance/mag/ra_genus_grade_mag_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

mag.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

mag.genus.plot

ggsave("plots/relative_abundance/mag/ra_genus_genus_mag_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


plot = ggplot(avgs_grouped, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

mag.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

mag.species.plot

ggsave("plots/relative_abundance/mag/ra_genus_species_mag_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


#Kraken -BACTERIA
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) var(x) > 0.001, TRUE)

df_ps.RA.filt <- psmelt(ps.RA.gen)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

# diet 
plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

krak.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

krak.diet.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_diet_krakbac_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

krak.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

krak.grade.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_grade_krakbac_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

krak.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

krak.genus.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_genus_krakbac_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

krak.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

krak.species.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_species_krakbac_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")




```

```{r RA_arc_fun_split_variance, eval=FALSE, include=FALSE}

#Kraken - archaea
ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))

ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) var(x) > 0.001, TRUE)

df_ps.RA.filt <- psmelt(ps.RA.gen)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

# diet 
plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

arc.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

arc.diet.plot

ggsave("plots/relative_abundance/archaea/ra_genus_diet_krakarc_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

arc.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

arc.grade.plot

ggsave("plots/relative_abundance/archaea/ra_genus_grade_krakarc_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

arc.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

arc.genus.plot

ggsave("plots/relative_abundance/archaea/ra_genus_genus_krakarc_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

arc.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

arc.species.plot

ggsave("plots/relative_abundance/archaea/ra_genus_species_krakarc_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#Kraken - Fungi
ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))

ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) var(x) > 0.001, TRUE)

df_ps.RA.filt <- psmelt(ps.RA.gen)

avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))
# diet 

plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

fun.diet.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

fun.diet.plot

ggsave("plots/relative_abundance/fungi/ra_genus_diet_krakfun_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


# terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

fun.grade.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

fun.grade.plot

ggsave("plots/relative_abundance/fungi/ra_genus_grade_krakfun_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

fun.genus.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

fun.genus.plot

ggsave("plots/relative_abundance/fungi/ra_genus_genus_krakfun_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#species 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Species, phylum, class, order, family, genus, OTU) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


plot = ggplot(avgs_grouped, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
width = 0.4, position = position_dodge(0.9))

fun.species.plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + facet_wrap(~genus, scales="free")

fun.species.plot

ggsave("plots/relative_abundance/fungi/ra_genus_species_krakfun_byvar_split.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


```

# Differential Genera
### DIET ###
### Is there a relationship between diet and specific genera in the guts of crabs?

### GRADE ###
### Is there a relationship between terrestrial grades and specific genera in the guts of crabs?

### PHYLOGENY ###
### Is there a taxonomic signal (in the crabs) when examining microbial community composition?

```{r}

# Initialize data frames to store KW and Dunn test p-values
RA_kw_pvals <- data.frame()
RA_dunn_pvals <- data.frame()

# MAG

# Transform counts to Relative Abundance (RA) and aggregate to genus level
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)

# Filter out genera with mean abundance < 0.01
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)

# Export from phyloseq to regular data frame
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Run Kruskal-Wallis test and store the p-values for grade
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Grade")
test.res.kw$DataSet <- "MAG"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

# Identify genera with significant differences in abundance for grade
mag.grade.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

# Run Dunn test for pairwise comparisons in genera with significant differences
dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Grade")
dunn.res$DataSet <- "MAG"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Run Kruskal-Wallis test and store the p-values for diet
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Diet_group")
test.res.kw$DataSet <- "MAG"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

# Identify genera with significant differences in abundance for diet
mag.diet.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

# Run Dunn test for pairwise comparisons in genera with significant differences
dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Diet_group")
dunn.res$DataSet <- "MAG"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Run Kruskal-Wallis test and store the p-values for genus
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Genus")
test.res.kw$DataSet <- "MAG"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

# Identify genera with significant differences in abundance at genus level
mag.genus.taxa =unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

# Run Dunn test for pairwise comparisons in genera with significant differences
dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Genus")
dunn.res$DataSet <- "MAG"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)


# Kraken - Bacteria
# Repeat save methods above to assess relative abundance of genera 

ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Testing terrestrial grade
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Grade")
test.res.kw$DataSet <- "Kraken - Bacteria"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakbac.grade.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Grade")
dunn.res$DataSet <- "Kraken - Bacteria"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing diet
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Diet_group")
test.res.kw$DataSet <- "Kraken - Bacteria"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakbac.diet.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Diet_group")
dunn.res$DataSet <- "Kraken - Bacteria"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing host genus
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Genus")
test.res.kw$DataSet <- "Kraken - Bacteria"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakbac.genus.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Genus")
dunn.res$DataSet <- "Kraken - Bacteria"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)


# Kraken - Archaea
# Repeat save methods above to assess relative abundance of genera 

ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Testing terrestrial grade
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Grade")
test.res.kw$DataSet <- "Kraken - Archaea"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakarc.grade.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

# dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Grade")
# dunn.res$DataSet <- "Kraken - Archaea"
# RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing diet
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Diet_group")
test.res.kw$DataSet <- "Kraken - Archaea"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakarc.diet.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Diet_group")
dunn.res$DataSet <- "Kraken - Archaea"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing host genus
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Genus")
test.res.kw$DataSet <- "Kraken - Archaea"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakarc.genus.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Genus")
dunn.res$DataSet <- "Kraken - Archaea"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)


# Kraken - Fungi
# Repeat save methods above to assess relative abundance of genera 

ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Testing terrestrial grade
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Grade")
test.res.kw$DataSet <- "Kraken - Fungi"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakfun.grade.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Grade")
dunn.res$DataSet <- "Kraken - Fungi"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing diet
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Diet_group")
test.res.kw$DataSet <- "Kraken - Fungi"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakfun.diet.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Diet_group")
dunn.res$DataSet <- "Kraken - Fungi"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)

# Testing host genus
test.res.kw <- get_kw_results_rel_abun(df_ps.RA.filt, "genus", "Genus")
test.res.kw$DataSet <- "Kraken - Fungi"
RA_kw_pvals <- bind_rows(test.res.kw, RA_kw_pvals)

krakfun.genus.taxa = unique(test.res.kw$Taxon[test.res.kw$padj < 0.05])

dunn.res <- get_dunn_results_rel_abun((df_ps.RA.filt %>% filter(genus %in% test.res.kw$Taxon[test.res.kw$padj < 0.05])), "genus", "Genus")
dunn.res$DataSet <- "Kraken - Fungi"
RA_dunn_pvals <- bind_rows(dunn.res, RA_dunn_pvals)


# Save results of statistical assessment of relative abundant taxa
#write.csv(RA_kw_pvals, "results/kw_relabun_pvals.csv", quote = FALSE, row.names = FALSE)
#write.csv(RA_dunn_pvals, "results/dunn_relabun_pvals.csv", quote = FALSE, row.names = FALSE)


```


```{r diffy_RA, fig.height=12,fig.width=16}

# MAG
# Transform counts to Relative Abundance (RA) and aggregate to genus level
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Plotting only genera with significant differences based on diet
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% mag.diet.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

# Create a bar plot with error bars
sig.mag.diet.plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), 
                width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

# Display and save the plot as a PNG
sig.mag.diet.plot
ggsave("plots/relative_abundance/mag/ra_genus_diet_mag_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# Repeat the above process for significant differences in genus abundance based on terrestrial grade and host genus
# The code and comments for these sections are analogous to the above section.

# Terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% mag.grade.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

sig.mag.grade.plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
                width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() +
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.mag.grade.plot
ggsave("plots/relative_abundance/mag/ra_genus_grade_mag_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# Host genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% mag.genus.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


sig.mag.genus.plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)),
                width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.mag.genus.plot
ggsave("plots/relative_abundance/mag/ra_genus_genus_mag_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


# Kraken - Bacteria
# Repeat the above process for plotting genera with significant differences across groups

ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Diet 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakbac.diet.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

sig.krakbac.diet.plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakbac.diet.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_diet_krakbac_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# Terrestrial grade 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Grade, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakbac.grade.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

sig.krakbac.grade.plot = ggplot(avgs_grouped, aes(x = Grade, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9))  + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "D")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakbac.grade.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_grade_krakbac_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# Host genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakbac.genus.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


sig.krakbac.genus.plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakbac.genus.plot

ggsave("plots/relative_abundance/bacteria/ra_genus_genus_krakbac_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


# Kraken - Archaea
# Repeat the above process for plotting genera with significant differences across groups

ps.RA <- transform_sample_counts(physeq_krak_nC_arc,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Diet 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakarc.diet.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

sig.krakarc.diet.plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakarc.diet.plot

ggsave("plots/relative_abundance/archaea/ra_genus_diet_krakarc_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

#genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakarc.genus.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


sig.krakarc.genus.plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakarc.genus.plot

ggsave("plots/relative_abundance/archaea/ra_genus_genus_krakarc_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")


# Kraken - Fungi
# Repeat the above process for plotting genera with significant differences across groups

ps.RA <- transform_sample_counts(physeq_krak_nC_fun,  function(x) x/sum(x))
ps.RA.gen <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
ps.RA.gen = filter_taxa(ps.RA.gen, function(x) mean(x) > 0.01, TRUE)
df_ps.RA.filt <- psmelt(ps.RA.gen)

# Diet 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Diet_group, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakfun.diet.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))

sig.krakfun.diet.plot = ggplot(avgs_grouped, aes(x = Diet_group, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9)) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "C")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakfun.diet.plot

ggsave("plots/relative_abundance/fungi/ra_genus_diet_krakfun_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")

# Host genus 
avgs_grouped <- df_ps.RA.filt %>% 
  group_by(Genus, phylum, class, order, family, genus, OTU) %>%
  filter(genus %in% krakfun.genus.taxa) %>%
  summarise(meanA =  100 * mean(Abundance),
            sdA = 100 * sd(Abundance), seA = 100 * se(Abundance))


sig.krakfun.genus.plot = ggplot(avgs_grouped, aes(x = Genus, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  geom_errorbar(aes(ymin = (meanA - seA), ymax = (meanA + seA)), width = 0.4, position = position_dodge(0.9))  + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) + 
  facet_wrap(~genus, scales="free")

sig.krakfun.genus.plot

ggsave("plots/relative_abundance/fungi/ra_genus_genus_krakfun_byvar_sig.png", dpi=300, device = "png", height = 12, width= 16, units = "in")



```



```{r eval=FALSE, include=FALSE}
# library(DESeq2)
# 
# #use non-RA data
# deseq.ps.its <- phyloseq_to_deseq2(physeq_krak_nC_bac, ~ Diet_group) 
# 
# #filter out low abundance asvs
# keep <- rowSums(counts(deseq.ps.its)) >= 10
# dds.ps.its <- deseq.ps.its[keep, ]
# 
# dds.its = DESeq(dds.ps.its, test = "Wald", fitType = "parametric")
# 
# resultsNames(dds.its)
# 
# alpha = 0.05
# 
# res.its <- results(dds.its, contrast = c("Wild.Captive",  "Captive", "Wild"), 
#         pAdjustMethod = "bonferroni")
# 
# # filter results by p-value
# res.its.alpha <- res.its[which(res.its$padj < alpha), ]
#   
# # Bind taxonomy to results
# res.alpha.its.tax = cbind(as(res.its.alpha, "data.frame"), as(tax_table(ps.its_decontam_NC_wild.captive)[rownames(res.its.alpha), 
#         ], "matrix"))
# 
# res.alpha.its.tax$ASV <- row.names(res.alpha.its.tax)
# 
# 
# res.alpha.its.tax <- res.alpha.its.tax %>%
#   mutate(Genus = ifelse(Genus == "Unclassified", ifelse(Family == "Unclassified", ifelse(Order == "Unclassified", ifelse(Class== "Unclassified", ifelse(Phylum == "Unclassified", "Unclassified Fungi", paste0("Unclassified ", as.character(Phylum))), paste0("Unclassified ", as.character(Class))), paste0("Unclassified ", as.character(Order))), paste0("Unclassified ", as.character(Family))), as.character(Genus)))%>%
#   mutate(ASV = paste0(as.character(Genus), " (", ASV, ")")) %>%
#   mutate(ASV = fct_reorder(ASV, log2FoldChange))
# 
# res.alpha.its.tax$OTU_ID <- rownames(res.alpha.its.tax)
# res.alpha.its.tax <- left_join(res.alpha.its.tax, guilds) %>%
#     mutate(`Trophic Mode` = ifelse(`Trophic Mode` == "-", "Unclassified", as.character(`Trophic Mode`)))
# 
# write.csv(res.alpha.its.tax, "../results/ITS/deseq.captivity.csv")
# 
# 
# # 
# #ggplot(res.16s.alpha.tax, aes(x = ASV, y = log2FoldChange, 
#         color = log2FoldChange)) + geom_point(size = 6) + theme(axis.text.x = element_text(angle = -70, 
#         hjust = 0, vjust = 0.5), plot.title = element_text(hjust = 0.5)) +   ylab(expression(paste("lo", 
#         g[2], " fold change"))) + coord_flip() + theme_minimal() +
#   geom_hline(yintercept=0, linetype="solid", color = "red") +
#   scale_color_gradientn(limits = c(-max(abs(res.alpha.its.tax$log2FoldChange), abs(res.16s.alpha.tax$log2FoldChange)), max(abs(res.alpha.its.tax$log2FoldChange),abs(res.16s.alpha.tax$log2FoldChange))), colors = rev(brewer.pal(11, "RdBu"))) + labs(color=expression(paste("lo", 
#         g[2], " fold change")))

```



# MICROBES OF INTEREST 
###  Do any microbial community members stand out in terms of lignocellulose decomposition ability?
###  Lignocellulose decomposition ability as determined from the literature

```{r sp_of_int, fig.height=12,fig.width=12}


# MAG 

# Transform counts to Relative Abundance (RA) at the genus level
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

# List of species of interest known from literature to degrade lignin
lignin_species <- c("Streptomyces", "Rhodococcus", "Enterobacter", "Pseudomonas", "Pseudomonas_A","Bacillus","Paenibacillus ", "Thermobifida","Clostridium", "Caldicellulosiruptor", "Citrobacter")

# Aggregate to the genus level 
ps.RA.sp <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
df_ps.RA.sp.filt <- psmelt(ps.RA.sp)

# Create a factor for species, specifying levels and labels
df_ps.RA.sp.filt$Species <- factor(df_ps.RA.sp.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))

# Group and summarize data for lignin-related species
avgs_grouped_lignin_species <- df_ps.RA.sp.filt %>% 
  group_by(Species, genus) %>%
  summarise(meanA = 100 * mean(Abundance)) %>%
  filter(genus %in% lignin_species)

# Create a bar plot
mag.plot = ggplot(avgs_grouped_lignin_species, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = position_dodge()) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  
  facet_wrap(~genus, scales = "free") 

# Display plot and save as PNG
mag.plot
ggsave("plots/relative_abundance/mag/mag_genus_lignin.png", dpi=300, device = "png", height = 6, width= 10, units = "in")


# Kraken - Bacteria

# Transform counts to Relative Abundance (RA) at the genus level
ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))

# Aggregate to the genus level 
ps.RA.sp <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
df_ps.RA.sp.filt <- psmelt(ps.RA.sp)

# Create a factor for species, specifying levels and labels
df_ps.RA.sp.filt$Species <- factor(df_ps.RA.sp.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))

# Group and summarize data for lignin-related species
avgs_grouped_lignin_species <- df_ps.RA.sp.filt %>% 
  group_by(Species, genus) %>%
  summarise(meanA = 100 * mean(Abundance)) %>%
  filter(genus %in% lignin_species)

# Create a bar plot
plot = ggplot(avgs_grouped_lignin_species, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = position_dodge()) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  
  facet_wrap(~genus, scales = "free") 

# Display and save as PNG
plot
ggsave("plots/relative_abundance/bacteria/krak_species_lignin.png", dpi=300, device = "png", height = 12, width= 12, units = "in")


```


```{r orig, include=FALSE, eval=FALSE}

# MAG 

# Transform counts to Relative Abundance (RA) at the genus level
ps.RA <- transform_sample_counts(physeq_coverM_nC,  function(x) x/sum(x))

# List of species of interest known from literature to degrade lignin
lignin_species <- c("Streptomyces", "Rhodococcus", "Enterobacter", "Pseudomonas", "Pseudomonas_A","Bacillus","Paenibacillus ", "Thermobifida","Clostridium", "Caldicellulosiruptor", "Citrobacter")

# Aggregate to the genus level 
ps.RA.sp <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)
df_ps.RA.sp.filt <- psmelt(ps.RA.sp)

# Create a factor for species, specifying levels and labels
df_ps.RA.sp.filt$Species <- factor(df_ps.RA.sp.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))

# Group and summarize data for lignin-related species
avgs_grouped_lignin_species <- df_ps.RA.sp.filt %>% 
  group_by(Species, genus) %>%
  summarise(meanA = 100 * mean(Abundance)) %>%
  filter(genus %in% lignin_species)

# Create a bar plot
mag.plot = ggplot(avgs_grouped_lignin_species, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = position_dodge()) + 
  theme_bw() + 
  theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + 
  xlab("") + 
  guides(fill = guide_legend(title = "Genus")) + 
  scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  
  facet_wrap(~genus, scales = "free") 

# Display plot and save as PNG
mag.plot
ggsave("plots/relative_abundance/mag/mag_genus_lignin.png", dpi=300, device = "png", height = 6, width= 10, units = "in")


## looking at within Tenericutes (renamed Mycoplasmatota)

#Phylum = "Tenericutes"

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.fam.filt <- psmelt(ps.RA.order)

avgs_grouped_ten <- df_ps.RA.fam.filt %>% 
  filter(phylum == "Tenericutes" | order == "Mycoplasmatales") %>%
  group_by(Species, order) %>%
  summarise(meanA = 100 * mean(Abundance)) 

plot = ggplot(avgs_grouped_ten, aes(x = Species, y = (meanA), fill = order)) + 
  geom_bar(stat = "identity", position = position_dodge())  

plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Order")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  facet_wrap(~order, scales = "free") 

plot

ggsave("plots/relative_abundance/mag/mag_family_in_Tenericutes.png", dpi=300, device = "png", height = 6, width= 10, units = "in")



# kraken


## plot only species of interest 

ps.RA <- transform_sample_counts(physeq_krak_nC_bac,  function(x) x/sum(x))


ps.RA.sp <- tax_glom(ps.RA, taxrank = "genus", NArm = FALSE)

df_ps.RA.sp.filt <- psmelt(ps.RA.sp)

df_ps.RA.sp.filt$Species <- factor(df_ps.RA.sp.filt$Species,
                                         levels = c("Petrolisthes armatus","Petrolisthes bolivarensis","Petrolisthes donadio","Petrolisthes haigae","Petrolisthes nobilii","Petrolisthes sp. new 1","Birgus latro","Tuerkayana celeste" ,"Tuerkayana magna", "Gecarcoidea humei" , "Gecarcoidea natalis","Geograpsus grayi","Ocypode ceratopthalmus"))

avgs_grouped_lignin_species <- df_ps.RA.sp.filt %>% 
  group_by(Species, genus) %>%
  summarise(meanA = 100 * mean(Abundance)) %>%
  filter(genus %in% lignin_species)

plot = ggplot(avgs_grouped_lignin_species, aes(x = Species, y = (meanA), fill = genus)) + 
  geom_bar(stat = "identity", position = position_dodge())  

plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Genus")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  facet_wrap(~genus, scales = "free") 

plot

ggsave("plots/relative_abundance/bacteria/krak_species_lignin.png", dpi=300, device = "png", height = 12, width= 12, units = "in")


## looking at within Tenericutes (renamed Mycoplasmatota)


#Phylum = "Tenericutes"

ps.RA.order <- tax_glom(ps.RA, taxrank = "order", NArm = FALSE)

df_ps.RA.ord.filt <- psmelt(ps.RA.order)

avgs_grouped_ten <- df_ps.RA.ord.filt %>% 
  mutate(Species = ifelse(Species == "New species 1", "Petrolitsthes sp.", as.character(Species))) %>%
  filter(phylum == "Tenericutes") %>%
  group_by(Species, order) %>%
  summarise(meanA = 100 * mean(Abundance)) 

plot = ggplot(avgs_grouped_ten, aes(x = Species, y = (meanA), fill = order)) + 
  geom_bar(stat = "identity", position = position_dodge())  

plot <- plot + theme_bw() + theme(text = element_text(size = 14)) +
  ylab("Mean Relative Abundance") + xlab("") + guides(fill = guide_legend(title = "Family")) + scale_fill_viridis_d(option = "H")  + 
  theme(axis.text.x = element_text(angle = -55, hjust = 0, vjust = 0.5)) +  facet_wrap(~order, scales = "free") 

plot

ggsave("plots/relative_abundance/bacteria/krak_order_in_Tenericutes.png", dpi=300, device = "png", height = 12, width= 16, units = "in")
```

