#!/bin/bash -l
#
#SBATCH --ntasks 16 #number cores
#SBATCH -J coverm
#SBATCH --mem=250G #memory
#SBATCH -p batch
#SBATCH -D /rhome/cassande/bigdata/eisenlab/crabs_eukrep/
#SBATCH -o logs/18_coverm.log
#SBATCH -e logs/18_coverm.log


#coverM installed in sourmash env
conda activate sourmash_to_phylo

IN=fastqs
BINS=bac_mags/fin_bins
OUT=coverm_results_fin
CPU=16

#coverm genome -c $IN/119_S24_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/119_S24_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/120_S25_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/120_S25_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/140_S26_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/140_S26_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/147_S27_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/147_S27_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/177_S28_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/177_S28_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/178_S29_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/178_S29_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/179_S31_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/179_S31_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/209_S30_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/209_S30_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/214_S32_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/214_S32_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/82_S21_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/82_S21_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/88_S22_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/88_S22_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/99_S23_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/99_S23_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLB_S18_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLB_S18_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLB_WG4_S19_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLB_WG4_S19_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLR_S17_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLR_S17_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLR_WG4_S20_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLR_WG4_S20_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLS_S16_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLS_S16_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC06_S1_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC06_S1_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC10_S2_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC10_S2_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC11_S3_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC11_S3_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC17_S4_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC17_S4_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM13_S9_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM13_S9_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM14_S10_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM14_S10_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM15_S11_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM15_S11_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM16_S12_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM16_S12_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GG01_S14_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GG01_S14_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GH01_S13_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GH01_S13_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN13_S5_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN13_S5_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN15_S6_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN15_S6_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN18_S7_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN18_S7_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN19_S8_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN19_S8_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/NEG_S33_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/NEG_S33_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/OC01_S15_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/OC01_S15_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz -d $BINS/ -x fa  -m relative_abundance rpkm --output-file $OUT/coverm_results.tsv -t $CPU 


coverm genome -c $IN/119_S24_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/119_S24_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/120_S25_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/120_S25_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/140_S26_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/140_S26_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/147_S27_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/147_S27_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/177_S28_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/177_S28_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/178_S29_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/178_S29_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/179_S31_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/179_S31_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/209_S30_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/209_S30_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/214_S32_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/214_S32_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/82_S21_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/82_S21_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/88_S22_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/88_S22_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/99_S23_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/99_S23_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLB_S18_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLB_S18_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLB_WG4_S19_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLB_WG4_S19_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLR_S17_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLR_S17_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLR_WG4_S20_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLR_WG4_S20_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/BLS_S16_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/BLS_S16_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC06_S1_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC06_S1_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC10_S2_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC10_S2_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC11_S3_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC11_S3_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DC17_S4_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DC17_S4_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM13_S9_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM13_S9_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM14_S10_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM14_S10_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM15_S11_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM15_S11_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/DM16_S12_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/DM16_S12_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GG01_S14_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GG01_S14_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GH01_S13_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GH01_S13_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN13_S5_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN13_S5_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN15_S6_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN15_S6_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN18_S7_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN18_S7_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/GN19_S8_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/GN19_S8_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/NEG_S33_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/NEG_S33_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz $IN/OC01_S15_L002_R1_001.fastq.gz.reads.filtered_1.fastq.gz $IN/OC01_S15_L002_R2_001.fastq.gz.reads.filtered_2.fastq.gz -d $BINS/ -x fa  -m relative_abundance rpkm --output-file $OUT/coverm_results_mincov0_aln75_id95.tsv -t $CPU  --min-read-aligned-percent 0.75 --min-read-percent-identity 0.95 --min-covered-fraction 0

